---
title: "Making Datasets figure"
author: Alexis Hucteau
date: "`r Sys.Date()`"
output:
    html_document:
      toc: yes
      toc_float: yes
      theme: united
editor_options:
  markdown:
    wrap: sentence
---

# Libraries

```{r, error=F, warning=F, echo=FALSE, include=F}
rm(DmIALite, DOLite, DOLiteTerm, HsIALite, MmIALite, RnIALite)

library(dplyr)
library(parallel)
library(ggplot2)

cores2use <- detectCores()
```

```{r}
Clinical_data_Koichi <- read.table("../../Datasets/Clinical_Koichi_data.tsv", sep = "\t", header = T)
```

```{r}
Pheno_RNAseq_data <- Clinical_data_Koichi$Baseline_RNAseq_data %>% na.omit %>% sapply(function(sample){
  tmp <- dplyr::filter(Clinical_data_Koichi, Baseline_RNAseq_data == sample)
  paste(tmp[3], tmp[11]) %>% unname
})

table_RNA <- table(Pheno_RNAseq_data)

data_set_RNA <- data.frame("Pheno" = names(table_RNA), 
                           "Number" = unname(table_RNA)) %>% .[c(1,3)]
```

```{r}
Pheno_DNAmethylation <- Clinical_data_Koichi$Baseline_Sample %>% na.omit %>% sapply(function(sample){
  tmp <- dplyr::filter(Clinical_data_Koichi, Baseline_Sample == sample)
  paste(tmp[3], tmp[11]) %>% unname
})

table_DNAmeth <- table(Pheno_DNAmethylation)

data_set_DNAmeth <- data.frame("Pheno" = names(table_DNAmeth), 
                           "Number" = unname(table_DNAmeth)) %>% .[c(1,3)]
```

```{r}
Proteomics_data <- read.table("../../Datasets/Proteomic/mIDH_data.tsv", sep = "\t", check.names = F, header = T, row.names = 1)
Pheno_classes <- data.frame("sample" = colnames(Proteomics_data), 
                            "mIDH" = ifelse(stringr::str_detect(colnames(Proteomics_data), "1_"), "mIDH1", "mIDH2"))

Samples_mIDH1 <- Pheno_classes %>% dplyr::filter(mIDH == "mIDH1") %>% .$sample
Samples_mIDH2 <- Pheno_classes %>% dplyr::filter(mIDH == "mIDH2") %>% .$sample

```

