---
title: "Chas_best_pchic"
author: Alexis Hucteau
date: "`r Sys.Date()`"
output:
    html_document:
      toc: yes
      toc_float: yes
      theme: united
editor_options:
  markdown:
    wrap: sentence
---


# Libraries

```{r, error=F, warning=F, echo=FALSE, include=F}
rm(DmIALite, DOLite, DOLiteTerm, HsIALite, MmIALite, RnIALite)

library(dplyr)
library(chaser)
library(stringr)
library(biomaRt)
library(gridExtra)
library(grid)
library(ggplot2)
library(ggforce)
library(data.table)
library(factoextra)
library(RColorBrewer)
library(pheatmap)
library(Hmisc)
library(parallel)

cores2use <- detectCores() -2
```

# Loading Data

```{r}
Clinical_patient_data <- read.table("../../Datasets/Clinical_Koichi_data.tsv", sep = "\t", header = T)
```


## Loading Gene expression

```{r}
RNAseq_Koichi <- read.csv("~/GitHub/Koichi_gene_expression_analyses_git/Koichi_gene_expression_analyses/DATA/RNAseq_parsed.csv", 
                          row.names = 1, header = T, check.names = F)

Phenos_Gene_expression_Data <- colnames(RNAseq_Koichi) %>% lapply(function(sample){
  fil <- Clinical_patient_data %>%
    dplyr::filter(Baseline_RNAseq_data == sample) 
  data.frame("IDH" = fil$IDH_isoform, "Cluster" = fil$Cluster, sample = fil$Baseline_RNAseq_data)
}) %>% data.table::rbindlist()

Samples_Gene_expression_Cluster_R <- Phenos_Gene_expression_Data %>% dplyr::filter(Cluster == "Cluster_R") %>% .$sample
Samples_Gene_expression_Cluster_NR <- Phenos_Gene_expression_Data %>% dplyr::filter(Cluster == "Cluster_NR") %>% .$sample
Samples_Gene_expression_mIDH1 <- Phenos_Gene_expression_Data %>% dplyr::filter(IDH == "IDH1") %>% .$sample
Samples_Gene_expression_mIDH2 <- Phenos_Gene_expression_Data %>% dplyr::filter(IDH == "IDH2") %>% .$sample

RNAseq_Cluster_R <- RNAseq_Koichi[, colnames(RNAseq_Koichi) %in% Samples_Gene_expression_Cluster_R]
RNAseq_Cluster_NR <- RNAseq_Koichi[, colnames(RNAseq_Koichi) %in% Samples_Gene_expression_Cluster_NR]
RNAseq_mIDH1 <- RNAseq_Koichi[, colnames(RNAseq_Koichi) %in% Samples_Gene_expression_mIDH1]
RNAseq_mIDH2 <- RNAseq_Koichi[, colnames(RNAseq_Koichi) %in% Samples_Gene_expression_mIDH2]
```

## Loading Methylation

```{r}
Anno_BMIQ <- read.table("../../Datasets/Annotation_EPIC_BMIQ.tsv", 
                        sep = "\t", header = T)

Phenos_Methylation_Data <- colnames(Anno_BMIQ) %>% mclapply(function(sample){
  fil <- Clinical_patient_data %>%
    dplyr::filter(Baseline_Sample == sample) 
  data.frame("IDH" = fil$IDH_isoform, "Cluster" = fil$Cluster, sample = fil$Baseline_Sample)
}, mc.cores = cores2use) %>% data.table::rbindlist()

Samples_Methylation_Cluster_R <- Phenos_Methylation_Data %>% dplyr::filter(Cluster == "Cluster_R") %>% .$sample
Samples_Methylation_Cluster_NR <- Phenos_Methylation_Data %>% dplyr::filter(Cluster == "Cluster_NR") %>% .$sample
Samples_Methylation_mIDH1 <- Phenos_Methylation_Data %>% dplyr::filter(IDH == "IDH1") %>% .$sample
Samples_Methylation_mIDH2 <- Phenos_Methylation_Data %>% dplyr::filter(IDH == "IDH2") %>% .$sample

Anno_Cluster_R <- cbind(Anno_BMIQ[, 1:3], Anno_BMIQ[, colnames(Anno_BMIQ) %in% Samples_Methylation_Cluster_R])
Anno_Cluster_NR <- cbind(Anno_BMIQ[, 1:3], Anno_BMIQ[, colnames(Anno_BMIQ) %in% Samples_Methylation_Cluster_NR])
Anno_mIDH1 <- cbind(Anno_BMIQ[, 1:3], Anno_BMIQ[, colnames(Anno_BMIQ) %in% Samples_Methylation_mIDH1])
Anno_mIDH2 <- cbind(Anno_BMIQ[, 1:3], Anno_BMIQ[, colnames(Anno_BMIQ) %in% Samples_Methylation_mIDH2])
```

## Loading Pchic

```{r}
load("/media/alexis/DATA/pchic.RData")
```

# Prepare Pchics chromatin network

```{r}
prepare_pchic <- function(cell_lines = NULL, minimum_interaction = 5, pchic = NULL){
  if(is.null(pchic)){
    load("/media/alexis/DATA/pchic.RData")
  }
  if(is.null(cell_lines)){
    cell_lines = c("Mon", "Mac0", "Mac1", "Mac2", "Neu", "MK", "EP", "Ery", "FoeT", "nCD4", "tCD4", "aCD4", "naCD4", "nCD8", "tCD8", "nB", "tB")
  }
  pchic <- data.frame(pchic[rowSums(pchic[,cell_lines] >= minimum_interaction) >= 1,c(1:3,5:8,10)]) %>% na.omit(.)
  pchic$baitChr <- paste0("chr", pchic$baitChr)
  pchic$oeChr <- paste0("chr", pchic$oeChr)
  po <- chaser::make_chromnet(pchic[,c(1:3, 5:7)])
  baits <- export(po, 'baits')
  chaser::subset_chromnet(po, method = "nodes", nodes1 = baits)
}
```

## Pchic chromatine per cell line

```{r, echo=F}
Myelo_cell_lines <- c("Mon", "Mac1", "Mac0", "Mac2", "MK", "Ery", "EP")

pchics <- mclapply(Myelo_cell_lines, function(cells){
  prepare_pchic(cells, pchic = pchic)
}, mc.cores = cores2use)
names(pchics) <- Myelo_cell_lines
```

# Chas analysis

## Calculates Random chas

```{r}
Make_chas_random_expression <- function(RNAseq, pchic){
  chas_random <- mclapply(names(pchics), function(cell_type){
    pp_exp <- chaser::load_features(pchics[[cell_type]],RNAseq,type='features_on_nodes',featnames = colnames(RNAseq), missingv=0)
    chas_random <- tryCatch(chaser::randomize(pp_exp, nrandom = 50, dist.match = T), error=function(e) NULL)
    lapply(chas_random, chas)
  }, mc.cores = cores2use)
  names(chas_random) <- Myelo_cell_lines
  chas_random
}
```

## Random Chas Expression

```{r, include = F, warning=F, echo = F}
Gene_exp_random_chas_Cluster_R <- Make_chas_random_expression(RNAseq_Cluster_R, pchics)
```





## Random Chas Methylation

```{r}

```
















