---
title: "BEATAML IDHms data analysis"
author: Alexis Hucteau
date: "`r Sys.Date()`"
output:
    html_document:
      toc: yes
      toc_float: yes
      theme: united
editor_options:
  markdown:
    wrap: sentence
---

```{r, error=FALSE}
rm(list = ls())

library(FactoMineR)
library(igraph)
library(limma)
library(viper)
library(stringr)
library(data.table)
library(dplyr)
library(RCy3)
library(factoextra)
library(pheatmap)
library(Hmisc)
library(matrixStats)
library(RColorBrewer)

"%ni%" <- Negate("%in%")
wd <- "~/GitHub/Thesis_paper/"
source("~/Core_scripts/msviper_functions.R")
source("~/Core_scripts/core_functions.R")
```

```{r}
RNAseqBEATAML <- read.table("../Datasets/Voom_BEATAML.tsv", sep = "\t", header = T)
Pheno_BEATAML <- sapply(colnames(RNAseqBEATAML), function(samples){
  IDH1 <- stringr::str_detect(samples, pattern = "mIDH1")
  IDH2 <- stringr::str_detect(samples, pattern = "mIDH2")
  ifelse(IDH1, "mIDH1", ifelse(IDH2, "mIDH2", "wtIDH"))
})
```

```{r}
RNAseqBEATAML <- RNAseqBEATAML[Pheno_BEATAML != "wtIDH"]
Pheno_BEATAML_IDH <- Pheno_BEATAML[Pheno_BEATAML != "wtIDH"]
```

```{r}
res.pca <- prcomp(t(RNAseqBEATAML))
p <- fviz_pca_ind(res.pca, label="all", habillage=Pheno_BEATAML_IDH,
                               addEllipses=T, ellipse.level=0.95)
p <- p + ggtitle("PCA gene expression")
p
png("../Results/DGEA/PCA_BEATAML_IDH_data.png")
p
dev.off()
```

```{r}
BEATAML_ARACNe_network <- read.table("../Results/DGEA/Networks_Beataml/network.txt", sep = "\t", header = T)
colnames(BEATAML_ARACNe_network)[1:2] <- c("tf", "target")

ref_mIDH1 <- Pheno_BEATAML_IDH == "mIDH1"
ref_mIDH2 <- Pheno_BEATAML_IDH == "mIDH2"

IDH_BEATAML_msvip <- run_msviper(RNAseqBEATAML, BEATAML_ARACNe_network, use_aracne = T, ref_mIDH1, ref_mIDH2, "mIDH1", "mIDH2", minsize = 4, ges.filter = T)
IDH_BEATAML_TF_actitity <- viper(RNAseqBEATAML, dorothea2viper_regulons(IDH_BEATAML_msvip$regulons))
IDH_BEATAML_msvip$mrs_table %>% dplyr::filter(pval < 0.05) %>% write.table("../Results/DGEA/Filtered_TF_Diff_activity_BEATAML_IDH.tsv", sep = "\t")
```

```{r}
plot(IDH_BEATAML_msvip$mrs, mrs = 50)
png("../Results/DGEA_Clusters_Clean/Cluster_BEATAML_IDH_msviplot.png", width = 1280, height = 720)
plot(IDH_BEATAML_msvip$mrs, mrs = 50)
dev.off()
```

```{r}
DEG_IDH_BEATAML <- Differential_analysis(Pheno_BEATAML_IDH, RNAseqBEATAML)$`mIDH1-mIDH2`
```

