---
title: "BEATAML data analysis"
author: Alexis Hucteau
date: "`r Sys.Date()`"
output:
    html_document:
      toc: yes
      toc_float: yes
      theme: united
editor_options:
  markdown:
    wrap: sentence
---

```{r, error=FALSE}
rm(list = ls())

suppressPackageStartupMessages({
  library(FactoMineR)
  library(igraph)
  library(limma)
  library(viper)
  library(stringr)
  library(data.table)
  library(dplyr)
  library(RCy3)
  library(factoextra)
  library(pheatmap)
  library(Hmisc)
  library(matrixStats)
  library(RColorBrewer)
})

"%ni%" <- Negate("%in%")
wd <- "~/GitHub/Thesis_paper/"
source("~/Core_scripts/msviper_functions.R")
```

```{r}
Combine_duplicated_gene_names_transcriptome <- function(transcriptome, colgenes){
  gene_list <- unique(transcriptome[,colgenes])
  transcriptome <- transcriptome %>%
    split(., .[colgenes]) %>%
    lapply(., function(x){
      l <- length(x[1,]) - 1
      cnames <- colnames(x)[c(1:l)]
      df <- x[,c(1:l)] %>%
        as.matrix(.) %>%
        colMaxs(.) %>% 
        data.frame(.) %>%
        t(.) %>%
        data.frame(.)
      colnames(df) <- cnames
      df
    }) %>%
    rbindlist(.) %>% 
    data.frame(., check.names = F)
  rownames(transcriptome) <- gene_list
  transcriptome
}
```


```{r}
BEATAML_raw_count <- read.table("/media/alexis/DATA/Thesis_paper_DATA/beataml_waves1to4_counts_dbgap.txt", sep = "\t", header = T)

Clinical_BEATAML_data <- readxl::read_xlsx("../Datasets/beataml_wv1to4_clinical.xlsx", sheet = 1)

Clinical_BEATAML_RNAseq_data <- dplyr::filter(Clinical_BEATAML_data, dbgap_rnaseq_sample %in% colnames(BEATAML_raw_count))

BEATAML_raw_count_clinicaled <- BEATAML_raw_count[,c(F,T,F,F, colnames(BEATAML_raw_count)[5:711] %in% Clinical_BEATAML_RNAseq_data$dbgap_rnaseq_sample)]

Pheno_IDH <- colnames(BEATAML_raw_count_clinicaled)[2:672] %>% sapply(function(sample){
  mutations <- dplyr::filter(Clinical_BEATAML_RNAseq_data, dbgap_rnaseq_sample == sample) %>% .$variantSummary
  stringr::str_detect(mutations, "IDH")
}) %>% unlist 

Pheno_IDH <- ifelse(is.na(Pheno_IDH), F, ifelse(Pheno_IDH, T, F))

BEATAML_IDHm_RawCounts <- BEATAML_raw_count_clinicaled[, c(T, Pheno_IDH)] 
BEATAML_IDHm_RawCounts <- BEATAML_IDHm_RawCounts[,c(2:98,1)]

BEATAML_IDHm_RawCounts_computed <- Combine_duplicated_gene_names_transcriptome(BEATAML_IDHm_RawCounts, colgenes = "display_label")
write.table(BEATAML_IDHm_RawCounts_computed, "../Datasets/BEATAML_IDHm_RawCounts_computed.tsv", sep = "\t", col.names = NA)
```

```{r}
Clinical_BEATAML_dnamethylation_filtered_IDHm$IDH1 <- sapply(Clinical_BEATAML_dnamethylation_filtered_IDHm$variantSummary, function(mutations){
  ifelse(stringr::str_detect(mutations, "IDH1"), "mIDH1", "wtIDH1")
})

Clinical_BEATAML_dnamethylation_filtered_IDHm$IDH2 <- sapply(Clinical_BEATAML_dnamethylation_filtered_IDHm$variantSummary, function(mutations){
  ifelse(stringr::str_detect(mutations, "IDH2"), "mIDH2", "wtIDH2")
})

Clinical_BEATAML_dnamethylation_filtered_IDHm$IDH <- paste(Clinical_BEATAML_dnamethylation_filtered_IDHm$IDH1, Clinical_BEATAML_dnamethylation_filtered_IDHm$IDH2, sep= "_") %>% stringr::str_remove(pattern = "wtIDH1_") %>% stringr::str_remove(pattern = "_wtIDH2")


Pheno_IDH_BEATAML <- sapply(colnames(BEATAML_IDHm_RawCounts_computed), function(samples){
  mutations <- dplyr::filter(Clinical_BEATAML_RNAseq_data, dbgap_rnaseq_sample == samples) %>% .$variantSummary
  IDH1 <- stringr::str_detect(mutations, pattern = "IDH1")
  IDH2 <- stringr::str_detect(mutations, pattern = "IDH2")
  dIDH <- IDH1 & IDH2
  ifelse(dIDH, "mIDH1&2", ifelse(IDH1, "mIDH1", "IDH2"))
})
Pheno_IDH_BEATAML %>% table
```



```{r}
res.pca <- prcomp(t(BEATAML_IDHm_RawCounts_computed))
p <- fviz_pca_ind(res.pca, label="all", habillage = Pheno_IDH_BEATAML,
                  addEllipses=T, ellipse.level=0.95)
p <- p + ggtitle("PCA gene expression")
p
png("../Results/DGEA/PCA_RNAseq_BEATAML_data.png")
p
dev.off()
```


```{r}
Voom_BEATAML <- voom(BEATAML_IDHm_RawCounts_computed)$E
write.table(Voom_BEATAML, "../Datasets/Voom_BEATAML.tsv", sep = "\t") 
```


```{r}
res.pca <- prcomp(t(Voom_BEATAML))
p <- fviz_pca_ind(res.pca, label="all", habillage = Pheno_IDH_BEATAML,
                  addEllipses=T, ellipse.level=0.95)
p <- p + ggtitle("PCA gene expression")
p
png("../Results/DGEA/PCA_voom_RNAseq_BEATAML_data.png")
p
dev.off()
```

```{r}
Make_heatmap <- function(DATA, Phenotype, method = "pearson", title, annotation_color, kmeans_k = NA, cuttree = NA, corr=T, TF=F) {
  annotation_for_heatmap <- data.frame(Phenotype = Phenotype$Phenotype)
  rownames(annotation_for_heatmap) <- colnames(DATA)
  if(corr){
    corr <- rcorr(as.matrix(DATA), type = method)$r
    colnames(corr) <- colnames(DATA)
    if(TF){
      color= rev(colorRampPalette(brewer.pal(n = 9, name = "RdBu"))(100))
    }else{
      color = colorRampPalette(brewer.pal(n = 9, name = "YlOrRd"))(100)
    }
  }else{
    corr=DATA
    color= rev(colorRampPalette(brewer.pal(n = 9, name = "RdBu"))(100))
  }
  
  title <- paste0(title, " ", method)
  heatmap <- pheatmap(corr, 
                      color = color,
                      annotation_col = annotation_for_heatmap,
                      annotation_colors = annotation_color,
                      legend = TRUE,
                      treeheight_row = 20,
                      main = title, 
                      fontsize = 10,
                      cutree_cols = cuttree
  )
  return(heatmap)
}
```

```{r}
Var_gene <- rowVars(as.matrix(Voom_BEATAML))
names(Var_gene) <- rownames(Voom_BEATAML)
```


```{r}
nb_genes <- 50
method <- "pearson"
Top_var_genes <- Var_gene[order(Var_gene, decreasing = T)] %>% .[1:nb_genes]
Top_Var_RNAseq <- Voom_BEATAML %>% .[rownames(.) %in% names(Top_var_genes),] 
ann_color_RNA <- list(
    Phenotype = c(IDHwt = "blue", mIDH1 = "red", mIDH2 = "orange"))

pheno_heatmap <- data.frame("IDHwt" = Pheno_BEATAML == "IDHwt", "mIDH1" = Pheno_BEATAML == "mIDH1", "mIDH2" = Pheno_BEATAML == "mIDH2")
pheno_heatmap$Phenotype <- Pheno_BEATAML
heat <- Make_heatmap(Top_Var_RNAseq, Phenotype = pheno_heatmap, title = paste0("Patient correlation of Top ", nb_genes, " variable Gene expression"), annotation_color = ann_color_RNA, method = method)

png(paste0("../Results/DGEA/Heatmap_GE_top_", nb_genes,"_corr_", method, "_BEATAML_data.png"))
heat
dev.off()

res.pca <- prcomp(t(Top_Var_RNAseq))
p <- fviz_pca_ind(res.pca, label="all", habillage=Pheno_BEATAML,
                               addEllipses=T, ellipse.level=0.95)
p <- p + ggtitle(paste0("PCA top ", nb_genes, " genes"))
p
png(paste0("../Results/DGEA/PCA_GE_top_", nb_genes,"_BEATAML_data.png"))
p
dev.off()
if(nb_genes < 100){
  heat2 <- Make_heatmap(Top_Var_RNAseq, Phenotype = pheno_heatmap, title = paste0("Patient correlation of Top ", nb_genes, " variable Gene expression"), annotation_color = ann_color_RNA, method = "", corr = F)
  png(paste0("../Results/DGEA/Heatmap_GE_top_", nb_genes,"_BEATAML_data.png"))
  heat2
  dev.off()
}
```



TF activity

```{r}
BEATAML_ARACNe_network <- read.table("../Results/DGEA/Networks_Beataml/network.txt", sep = "\t", header = T)
colnames(BEATAML_ARACNe_network)[1:2] <- c("tf", "target")

ref_wtIDH <- Pheno_BEATAML == "IDHwt"
ref_mIDH <- stringr::str_detect(Pheno_BEATAML, pattern = "mIDH")

IDH_BEATAML_msvip <- run_msviper(Voom_BEATAML, BEATAML_ARACNe_network, use_aracne = T, ref_wtIDH, ref_mIDH, "wtIDH", "mIDH", minsize = 4, ges.filter = T)
IDH_BEATAML_TF_actitity <- viper(Voom_BEATAML, dorothea2viper_regulons(IDH_BEATAML_msvip$regulons))
```

Heatmap

```{r}
Var_TF <- rowVars(as.matrix(IDH_BEATAML_TF_actitity))
names(Var_TF) <- rownames(IDH_BEATAML_TF_actitity)
```


```{r}
nb_TF <- 10
method <- "pearson"
Top_var_TF <- Var_TF[order(Var_TF, decreasing = T)] %>% .[1:nb_TF]
Top_Var_TF <- IDH_BEATAML_TF_actitity %>% .[rownames(.) %in% names(Top_var_TF),] 
heat <- Make_heatmap(Top_Var_TF, Phenotype = pheno_heatmap, title = paste0("Patient correlation of Top ", nb_TF, " variable TF"), annotation_color = ann_color_RNA, method = method,TF=T)

png(paste0("../Results/DGEA/Heatmap_TF_top_", nb_TF,"_corr_", method, "_BEATAML_data.png"))
heat
dev.off()

res.pca <- Top_Var_TF %>% as.data.frame %>% t() %>% prcomp

p <- fviz_pca_ind(res.pca, label="all", habillage=pheno_heatmap,
                               addEllipses=T, ellipse.level=0.95)
p <- p + ggtitle(paste0("PCA top ", nb_TF, " TF"))
p
png(paste0("../Results/DGEA/PCA_GE_top_", nb_TF,"_TF_BEATAML_data.png"))
p
dev.off()
heat2 <- Make_heatmap(Top_Var_TF, Phenotype = pheno_heatmap, title = paste0("Patient correlation of Top ", nb_TF, " variable TF"), annotation_color = ann_color_RNA, method = "", corr=F)

png(paste0("../Results/DGEA/Heatmap_TF_top_", nb_TF,"_BEATAML_data.png"))
heat2
dev.off()
```


```{r}
res.pca <- IDH_BEATAML_TF_actitity %>% as.data.frame %>% prcomp

p <- fviz_pca_ind(res.pca, label="all",
                               addEllipses=F, ellipse.level=0.95)
p <- p + ggtitle("PCA TF activity")
p
png("../Results/DGEA/Heatmap_PCA_TF_BEATAML_data.png")
p
dev.off()
```