---
title: "Chromatine_network_making"
author: Alexis Hucteau
date: "`r Sys.Date()`"
output:
    html_document:
      toc: yes
      toc_float: yes
      theme: united
editor_options:
  markdown:
    wrap: sentence
---

```{r}
rm(list = ls())
```

# Libraries

```{r}
  library(dplyr)
  library(RCy3)
  library(igraph)
  library(data.table)
  library(GenomicRanges)
  library(data.table)

"%ni%" <- Negate("%in%")
```

# dCPGs

## Clusters

```{r}
dCPGs_Clusters <- read.table("../Results/DNAmethylation/Dcpgs_ClusterNR_R.tsv", sep = "\t", header = T)
Anno_EPIC <- read.table("/media/alexis/DATA/Illumina_manifest/infinium-methylationepic.tsv", sep = "\t", header = T)
dCPGs_Clusters_annotated <- merge(dCPGs_Clusters, Anno_EPIC, by.x = "ID", by.y = "Name")
dCPGs_Clusters_annotated$Strand <- ifelse(dCPGs_Clusters_annotated$Strand == "F", "+", "-")
dCPGs_filtered_Clusters_annotated <- dplyr::filter(dCPGs_Clusters_annotated, P.Value < 0.05)
dCPGs_filtered_Clusters_annotated$abs_logFC <- abs(dCPGs_filtered_Clusters_annotated$logFC)
dCPGs_filtered_Clusters_annotated_ordered <- dCPGs_filtered_Clusters_annotated[order(dCPGs_filtered_Clusters_annotated$abs_logFC, decreasing = T),] %>% .[1:5000,]
dCPGs_Clusters_GRanges <- GRanges(
  seqnames = paste0("chr", dCPGs_filtered_Clusters_annotated_ordered$CHR),
  ranges = IRanges(dCPGs_filtered_Clusters_annotated_ordered$MAPINFO, dCPGs_filtered_Clusters_annotated_ordered$MAPINFO +1),
  strand = dCPGs_filtered_Clusters_annotated_ordered$Strand,
  CpG = dCPGs_filtered_Clusters_annotated_ordered$ID
)
```

## NR R




## IDH1 IDH2

```{r}
dCPGs_IDHm <- read.table("../Results/DNAmethylation/dCpGs_KOICHI_IDHm.tsv", sep = "\t", header = T, row.names = 1)
dCPGs_IDHm_annotated <- merge(dCPGs_IDHm, Anno_EPIC, by.x = "ID", by.y = "Name")
dCPGs_IDHm_annotated$Strand <- ifelse(dCPGs_IDHm_annotated$Strand == "F", "+", "-")
dCPGs_filtered_IDHm_annotated <- dplyr::filter(dCPGs_IDHm_annotated, P.Value < 0.05)
dCPGs_filtered_IDHm_annotated$abs_logFC <- abs(dCPGs_filtered_IDHm_annotated$logFC)
dCPGs_filtered_IDHm_annotated_ordered <- dCPGs_filtered_IDHm_annotated[order(dCPGs_filtered_IDHm_annotated$abs_logFC, decreasing = T),] %>% .[1:5000,]
dCPGs_IDHm_GRanges <- GRanges(
  seqnames = paste0("chr", dCPGs_filtered_IDHm_annotated_ordered$CHR),
  ranges = IRanges(dCPGs_filtered_IDHm_annotated_ordered$MAPINFO, dCPGs_filtered_IDHm_annotated_ordered$MAPINFO +1),
  strand = dCPGs_filtered_IDHm_annotated_ordered$Strand,
  CpG = dCPGs_filtered_IDHm_annotated_ordered$ID
)
```

# Promoter Capture Hi-C

## Myelo cell lines

```{r}
load("/media/alexis/DATA/pchic.RData")
cell_lines = c("Mon", "Mac1", "Mac0", "Mac2", "MK", "Ery", "EP")
pchic <- data.frame(pchic[rowSums(pchic[,cell_lines] >= 5) >= 1, 1:10]) %>% na.omit(.)
pchic_bed <- pchic
colnames(pchic_bed) <- rep(c("chr", "start", "end", "ID", "Name"), 2)
pchic_bed <- rbind(pchic_bed[,1:5], pchic_bed[,6:10])
Pchic_edges <- pchic[,c(4,9)]
colnames(Pchic_edges) <- c("source", "target")
pchic_Granges <- GRanges(seqnames = paste0("chr", pchic_bed$chr), 
                         ranges = IRanges(start = pchic_bed$start, end = pchic_bed$end), 
                         ID = pchic_bed$ID)
```

## HSC

```{r}
HSC_Pchic_cleaned <- read.table("../Datasets/HSC_Pchic_cleaned_gene_named.tsv", sep = "\t", header = T)
HSC_pchic_bed <- HSC_Pchic_cleaned[,c(1,3:6, 2, 7:10)]
colnames(HSC_pchic_bed) <- rep(c("ID", "chr", "start", "end", "Name"), 2)
HSC_pchic_bed <- rbind(HSC_pchic_bed[,c(1:5)], HSC_pchic_bed[,c(6:10)]) %>% 
  unique
HSC_Pchic_edges <- HSC_Pchic_cleaned[,1:2] %>% 
  unique
colnames(HSC_Pchic_edges) <- c("source", "target")
HSC_pchic_bed$chr <- paste0("chr", HSC_pchic_bed$chr)
HSC_pchic_Granges <- GRanges(seqnames = HSC_pchic_bed$chr, 
                         ranges = IRanges(start = HSC_pchic_bed$start, end = HSC_pchic_bed$end), 
                         ID = HSC_pchic_bed$ID,
                         Name = HSC_pchic_bed$Name)
```

# Overlaps

## Pchic

### IDHm

```{r}
overlaps <- findOverlaps(pchic_Granges, dCPGs_IDHm_GRanges)
dCPGs_IDHm_pchic <- data.frame(mcols(pchic_Granges[queryHits(overlaps),]),
                          data.frame(mcols(dCPGs_IDHm_GRanges[subjectHits(overlaps),]))) %>% unique()

pchic_net_mIDH <- dplyr::filter(Pchic_edges, source %in% dCPGs_IDHm_pchic$ID | target %in% dCPGs_IDHm_pchic$ID)

dCPGs_filtered_mIDH_annotated <- merge(dCPGs_filtered_IDHm_annotated_ordered, dCPGs_IDHm_pchic, by.x = "ID", by.y = "CpG")
colnames(dCPGs_filtered_mIDH_annotated)[12] <- "fragment_ID"
write.table(dCPGs_filtered_mIDH_annotated, "../Results/DNAmethylation/dCPGs_mIDH_pchic.tsv", sep = "\t", row.names = F)
write.table(pchic_net_mIDH, "../Results/Chromatine_part/pchic_dCPGs_mIDH.tsv", sep = "\t", row.names = F)
```

```{r}
pchic_gene_network_mIDH <- pchic[,1:5] %>% unique() %>% dplyr::filter(baitID %in% c(pchic_net_mIDH$source, pchic_net_mIDH$target))
pchic_gene_network_mIDH <- tidyr::separate_rows(pchic_gene_network_mIDH, baitName, sep = ";")
pchic_gene_network_mIDH <- unique(pchic_gene_network_mIDH)
pchic_gene_network_mIDH <- pchic_gene_network_mIDH[,c("baitID", "baitName")]
colnames(pchic_gene_network_mIDH) <- c("source", "target")

pchic_gene_network_mIDH <- dplyr::filter(pchic_gene_network_mIDH, target %ni% c("snoU13", "Y_RNA"))
```

### Clusters

```{r}
overlaps <- findOverlaps(pchic_Granges, dCPGs_Clusters_GRanges)
dCPGs_Clusters_pchic <- data.frame(mcols(pchic_Granges[queryHits(overlaps),]),
                          data.frame(mcols(dCPGs_Clusters_GRanges[subjectHits(overlaps),]))) %>% unique()

pchic_net <- dplyr::filter(Pchic_edges, source %in% dCPGs_Clusters_pchic$ID | target %in% dCPGs_Clusters_pchic$ID)

dCPGs_filtered_Clusters_annotated <- merge(dCPGs_filtered_Clusters_annotated_ordered, dCPGs_Clusters_pchic, by.x = "ID", by.y = "CpG")
colnames(dCPGs_filtered_Clusters_annotated)[12] <- "fragment_ID"
write.table(dCPGs_filtered_Clusters_annotated, "../Results/DNAmethylation/dCPGs_Clusters_pchic.tsv", sep = "\t", row.names = F)
write.table(pchic_net, "../Results/Chromatine_part/pchic_dCPGs_Clusters.tsv", sep = "\t", row.names = F)
```

```{r}
pchic_gene_network <- pchic[,1:5] %>% unique() %>% dplyr::filter(baitID %in% c(pchic_net$source, pchic_net$target))
pchic_gene_network <- tidyr::separate_rows(pchic_gene_network, baitName, sep = ";")
pchic_gene_network <- unique(pchic_gene_network)
pchic_gene_network <- pchic_gene_network[,c("baitID", "baitName")]
colnames(pchic_gene_network) <- c("source", "target")

pchic_gene_network <- dplyr::filter(pchic_gene_network, target %ni% c("snoU13", "Y_RNA"))
```

## HSC

### IDHm

```{r}
overlaps <- findOverlaps(HSC_pchic_Granges, dCPGs_IDHm_GRanges)
dCPGs_mIDH_HSC <- data.frame(mcols(HSC_pchic_Granges[queryHits(overlaps),]),
                          data.frame(mcols(dCPGs_IDHm_GRanges[subjectHits(overlaps),]))) %>% unique()

HSC_net_mIDH <- dplyr::filter(HSC_Pchic_edges, source %in% dCPGs_mIDH_HSC$ID | target %in% dCPGs_mIDH_HSC$ID)

dCPGs_filtered_mIDH_HSC_annotated <- merge(dCPGs_filtered_Clusters_annotated_ordered, dCPGs_mIDH_HSC, by.x = "ID", by.y = "CpG")
colnames(dCPGs_filtered_mIDH_HSC_annotated)[12] <- "fragment_ID"
write.table(dCPGs_filtered_mIDH_HSC_annotated, "../Results/DNAmethylation/dCPGs_mIDH_HSC.tsv", sep = "\t", row.names = F)
write.table(HSC_net_mIDH, "../Results/Chromatine_part/HSC_dCPGs_mIDH.tsv", sep = "\t", row.names = F)
```

```{r}
HSC_gene_network <- unique(HSC_Pchic_cleaned)
HSC_gene_network <- HSC_gene_network[,c("bait_ID", "bait_name")]
colnames(HSC_gene_network) <- c("source", "target")

HSC_gene_network <- dplyr::filter(HSC_gene_network, target %ni% c("snoU13", "Y_RNA")) %>%
  na.omit %>%
  unique
```

### Clusters

```{r}
overlaps <- findOverlaps(HSC_pchic_Granges, dCPGs_Clusters_GRanges)
dCPGs_Clusters_HSC <- data.frame(mcols(HSC_pchic_Granges[queryHits(overlaps),]),
                          data.frame(mcols(dCPGs_Clusters_GRanges[subjectHits(overlaps),]))) %>% unique()

HSC_net <- dplyr::filter(HSC_Pchic_edges, source %in% dCPGs_Clusters_HSC$ID | target %in% dCPGs_Clusters_HSC$ID)

dCPGs_filtered_Clusters_HSC_annotated <- merge(dCPGs_filtered_Clusters_annotated_ordered, dCPGs_Clusters_HSC, by.x = "ID", by.y = "CpG")
colnames(dCPGs_filtered_Clusters_HSC_annotated)[12] <- "fragment_ID"
write.table(dCPGs_filtered_Clusters_HSC_annotated, "../Results/DNAmethylation/dCPGs_Clusters_HSC.tsv", sep = "\t", row.names = F)
write.table(HSC_net, "../Results/Chromatine_part/HSC_dCPGs_Clusters.tsv", sep = "\t", row.names = F)
```

```{r}
HSC_gene_network <- unique(HSC_Pchic_cleaned)
HSC_gene_network <- HSC_gene_network[,c("bait_ID", "bait_name")]
colnames(HSC_gene_network) <- c("source", "target")

HSC_gene_network <- dplyr::filter(HSC_gene_network, target %ni% c("snoU13", "Y_RNA")) %>%
  na.omit %>%
  unique
```

# Make network readable

## Functions

```{r}
Interconnect_all_node_from_vector <- function(vec){
  len <- length(vec)
  source <- sapply(1:len, function(element){
    rep(vec[element], len-element)
  }) %>% unlist()
  target <- sapply(1:(len-1), function(element){
    vec[(1+element):len]
  }) %>% unlist()
  data.frame("source" = source,
             "target" = target)
}

Replace_fragments_promoter_to_gene_names <- function(pchic_network, fragment_gene_annotation){
  res <- lapply(1:nrow(pchic_network), function(n_row){
    fragment_A_gene <- dplyr::filter(fragment_gene_annotation, source == pchic_network[n_row, "source"]) %>% .$target
    if (length(fragment_A_gene) == 0){
      fragment_A_gene <- pchic_network[n_row, "source"]
    }
    fragment_B_gene <- dplyr::filter(fragment_gene_annotation, source == pchic_network[n_row, "target"]) %>% .$target
    if (length(fragment_B_gene) == 0){
      fragment_B_gene <- pchic_network[n_row, "target"]
    }
    vec <- unique(c(fragment_A_gene, fragment_B_gene))
    
    if(n_row %% floor(nrow(pchic_network)/100) == 0){
      message(paste0(n_row %/% floor(nrow(pchic_network)/100), "%!"))
    }
    if(length(vec) == 1){
      data.frame("source" = pchic_network[n_row, "source"], "target"= pchic_network[n_row, "target"])
    }else{
      Interconnect_all_node_from_vector(vec)
    }
  })

  res <- rbindlist(res)
  unique(res)
}
```

## PCHIC

### IDHm

```{r}
Promoter_fragment_mIDH <- Replace_fragments_promoter_to_gene_names(pchic_net_mIDH, pchic_gene_network)
c(Promoter_fragment_mIDH$source, Promoter_fragment_mIDH$target) %>% unique %>% length
Promoter_fragment_mIDH %>% write.table("../Results/Chromatine_part/Chromatine_networks/Pchic_KOICHI_mIDH.tsv", sep = "\t", quote = F, row.names = F)
```

```{r}
Pchic_Fragments_features_mIDH <- merge(dCPGs_filtered_mIDH_annotated, pchic_gene_network, by.x = "fragment_ID", by.y = "source", all.x = T)
Pchic_Fragments_features_mIDH$target <- sapply(1:nrow(Pchic_Fragments_features_mIDH), function(fragment){
  if(is.na(Pchic_Fragments_features_mIDH[fragment,"target"])){
    Pchic_Fragments_features_mIDH[fragment,"fragment_ID"]
  }else{
    Pchic_Fragments_features_mIDH[fragment, "target"]
  }
})

Pchic_Fragments_features_mIDH_promoter_fragments <- dplyr::filter(Pchic_Fragments_features_mIDH, target %in% Promoter_fragment_mIDH$source | target %in% Promoter_fragment_mIDH$target)

Pchic_Fragments_features_mIDH_promoter_fragments <- split(Pchic_Fragments_features_mIDH_promoter_fragments, Pchic_Fragments_features_mIDH_promoter_fragments$target)

Pchic_Fragments_features_mIDH_promoter_fragments <- lapply(names(Pchic_Fragments_features_mIDH_promoter_fragments), function(gene){
  data.frame("name" = gene, 
             "logFC" = mean(Pchic_Fragments_features_mIDH_promoter_fragments[[gene]]$logFC), 
             "P.Value" = max(Pchic_Fragments_features_mIDH_promoter_fragments[[gene]]$P.Value))
}) %>% rbindlist()

write.table(Pchic_Fragments_features_mIDH_promoter_fragments, "../Results/Chromatine_part/Chromatine_features_datasets/Pchic_KOICHI_mIDH.tsv", sep = "\t", quote = F, row.names = F)
```

### Clusters

```{r}
Promoter_fragment <- Replace_fragments_promoter_to_gene_names(pchic_net, pchic_gene_network)
c(Promoter_fragment$source, Promoter_fragment$target) %>% unique %>% length
Promoter_fragment %>% write.table("../Results/Chromatine_part/Chromatine_networks/Pchic_Koichi_Clusters.tsv", sep = "\t", quote = F, row.names = F)
```

```{r}
Fragments_features <- merge(dCPGs_filtered_Clusters_annotated, pchic_gene_network, by.x = "fragment_ID", by.y = "source", all.x = T)
Fragments_features$target <- sapply(1:nrow(Fragments_features), function(fragment){
  if(is.na(Fragments_features[fragment,"target"])){
    Fragments_features[fragment,"fragment_ID"]
  }else{
    Fragments_features[fragment, "target"]
  }
})

Fragments_features_promoter_fragments <- dplyr::filter(Fragments_features, target %in% Promoter_fragment$source | target %in% Promoter_fragment$target)

Fragments_features_promoter_fragments <- split(Fragments_features_promoter_fragments, Fragments_features_promoter_fragments$target)

Fragments_features_promoter_fragments <- lapply(names(Fragments_features_promoter_fragments), function(gene){
  data.frame("name" = gene, 
             "logFC" = mean(Fragments_features_promoter_fragments[[gene]]$logFC), 
             "P.Value" = max(Fragments_features_promoter_fragments[[gene]]$P.Value))
}) %>% rbindlist()

write.table(Fragments_features_promoter_fragments, "../Results/Chromatine_part/Chromatine_features_datasets/Pchic_KOICHI_clusters.tsv", sep = "\t", quote = F, row.names = F)
```

## HSC

### IDHm

```{r}
Promoter_HSC_fragment_mIDH <- Replace_fragments_promoter_to_gene_names(HSC_net_mIDH, HSC_gene_network)
c(Promoter_HSC_fragment_mIDH$source, Promoter_HSC_fragment_mIDH$target) %>% unique %>% length
Promoter_HSC_fragment_mIDH %>%
  write.table("../Results/Chromatine_part/Chromatine_networks/HSC_KOICHI_mIDH.tsv", 
                                           sep = "\t", 
                                           quote = F, 
                                           row.names = F)
```

```{r}
HSC_Fragments_features_mIDH <- merge(dCPGs_filtered_mIDH_HSC_annotated, HSC_gene_network, by.x = "fragment_ID", by.y = "source", all.x = T) %>%
  unique
HSC_Fragments_features_mIDH$target <- sapply(1:nrow(HSC_Fragments_features_mIDH), function(fragment){
  if(is.na(HSC_Fragments_features_mIDH[fragment,"target"])){
    HSC_Fragments_features_mIDH[fragment,"fragment_ID"]
  }else{
    HSC_Fragments_features_mIDH[fragment, "target"]
  }
})

HSC_Fragments_features_promoter_fragments_mIDH <- dplyr::filter(HSC_Fragments_features_mIDH, target %in% Promoter_HSC_fragment_mIDH$source | target %in% Promoter_HSC_fragment_mIDH$target)

HSC_Fragments_features_promoter_fragments_mIDH <- split(HSC_Fragments_features_promoter_fragments_mIDH, HSC_Fragments_features_promoter_fragments_mIDH$target)

HSC_Fragments_features_promoter_fragments_mIDH <- lapply(names(HSC_Fragments_features_promoter_fragments_mIDH), function(gene){
  data.frame("name" = gene, 
             "logFC" = mean(HSC_Fragments_features_promoter_fragments_mIDH[[gene]]$logFC), 
             "P.Value" = max(HSC_Fragments_features_promoter_fragments_mIDH[[gene]]$P.Value))
}) %>% rbindlist()

write.table(HSC_Fragments_features_promoter_fragments_mIDH, "../Results/Chromatine_part/Chromatine_features_datasets/HSC_KOICHI_mIDH.tsv", sep = "\t", quote = F, row.names = F)
```

### Clusters

```{r}
Promoter_HSC_fragment <- Replace_fragments_promoter_to_gene_names(HSC_net, HSC_gene_network)
c(Promoter_HSC_fragment$source, Promoter_HSC_fragment$target) %>% unique %>% length
Promoter_HSC_fragment %>% write.table("../Results/Chromatine_part/Chromatine_networks/HSC_KOICHI_Clusters.tsv", sep = "\t", quote = F, row.names = F)
```

```{r}
HSC_Fragments_features <- merge(dCPGs_filtered_Clusters_HSC_annotated, HSC_gene_network, by.x = "fragment_ID", by.y = "source", all.x = T) %>%
  unique
HSC_Fragments_features$target <- sapply(1:nrow(HSC_Fragments_features), function(fragment){
  if(is.na(HSC_Fragments_features[fragment,"target"])){
    HSC_Fragments_features[fragment,"fragment_ID"]
  }else{
    HSC_Fragments_features[fragment, "target"]
  }
})

HSC_Fragments_features_promoter_fragments <- dplyr::filter(HSC_Fragments_features, target %in% Promoter_HSC_fragment$source | target %in% Promoter_HSC_fragment$target)

HSC_Fragments_features_promoter_fragments <- split(HSC_Fragments_features_promoter_fragments, HSC_Fragments_features_promoter_fragments$target)

HSC_Fragments_features_promoter_fragments <- lapply(names(HSC_Fragments_features_promoter_fragments), function(gene){
  data.frame("name" = gene, 
             "logFC" = mean(HSC_Fragments_features_promoter_fragments[[gene]]$logFC), 
             "P.Value" = max(HSC_Fragments_features_promoter_fragments[[gene]]$P.Value))
}) %>% rbindlist()

write.table(HSC_Fragments_features_promoter_fragments, "../Results/Chromatine_part/Chromatine_features_datasets/HSC_KOICHI_clusters.tsv", sep = "\t", quote = F, row.names = F)
```
