---
title: "HSC_CD34_Pchic_analysis"
author: Alexis Hucteau
date: "`r Sys.Date()`"
output:
    html_document:
      toc: yes
      toc_float: yes
      theme: united
editor_options:
  markdown:
    wrap: sentence
---


```{r}
rm(list = ls())

suppressPackageStartupMessages({
  library(dplyr)
  library(GenomicRanges)
  library(biomaRt)
  library(data.table)
})

source("~/Core_scripts/core_functions.R")
source("~/GitHub/Multiplex_DNAmet_PPI_Chrom_Coexp/Make_koichi_factor.R")
```


```{r}
HSC_Pchic_file <- read.table("../Datasets/HSC_fetal_merge_cutoff_5.ibed", sep = "\t", header = T)
load("/media/alexis/DATA/pchic.RData")
HSC_Pchic_annotations <- read.table("../Datasets/baits_coordinates_annotated_ensembl_transcript_id_GRCh38_104.bed", sep = "\t", header = T)
```

```{r}
DMP_Clusters <- read.table("../Results/DNAmethylation/DMP_Cluster_NR_R.tsv", sep = "\t", header = T, row.names = 1)
dCPGs_Clusters <- read.table("../Results/DNAmethylation/Dcpgs_ClusterNR_R.tsv", sep = "\t", header = T)
Anno_EPIC <- read.table("/media/alexis/DATA/Illumina_manifest/infinium-methylationepic.tsv", sep = "\t", header = T)
dCPGs_Clusters_annotated <- merge(dCPGs_Clusters, Anno_EPIC, by.x = "ID", by.y = "Name")
dCPGs_Clusters_annotated$Strand <- ifelse(dCPGs_Clusters_annotated$Strand == "F", "+", "-")
dCPGs_filtered_Clusters_annotated <- dplyr::filter(dCPGs_Clusters_annotated, P.Value < 0.05 & abs(logFC) > 0.3)
dCPGs_Clusters_GRanges <- GRanges(
  seqnames = dCPGs_filtered_Clusters_annotated$CHR,
  ranges = IRanges(dCPGs_filtered_Clusters_annotated$MAPINFO, dCPGs_filtered_Clusters_annotated$MAPINFO +1),
  strand = dCPGs_filtered_Clusters_annotated$Strand,
  CpG = dCPGs_filtered_Clusters_annotated$ID
)
```


```{r}
HSC_Pchic_file$bait_ID <- paste(HSC_Pchic_file$bait_chr, HSC_Pchic_file$bait_start, HSC_Pchic_file$bait_end, sep = "_")
HSC_Pchic_file$Oe_ID <- paste(HSC_Pchic_file$otherEnd_chr, HSC_Pchic_file$otherEnd_start, HSC_Pchic_file$otherEnd_end, sep = "_")
HSC_Pchic_annotations$ID_tmp <- paste(HSC_Pchic_annotations$chr, HSC_Pchic_annotations$start, HSC_Pchic_annotations$end, sep = "_")

HSC_Pchic_file_test <- merge(HSC_Pchic_file, HSC_Pchic_annotations[,c("ID", "ID_tmp")], by.x = "bait_ID", by.y = "ID_tmp", all.x = T)
HSC_Pchic_file_test <- merge(HSC_Pchic_file_test, HSC_Pchic_annotations[,c("ID", "ID_tmp")], by.x = "Oe_ID", by.y = "ID_tmp", all.x = T)
colnames(HSC_Pchic_file_test)[13:14] <- c("bait_ID", "Other_end_ID")
i <- max(na.omit(HSC_Pchic_file_test$Other_end_ID))
HSC_Pchic_file_test_OE <- HSC_Pchic_file_test[is.na(HSC_Pchic_file_test$Other_end_ID),]
HSC_Pchic_file_test_OE$Other_end_ID <- sapply(1:nrow(HSC_Pchic_file_test_OE), function(row){
  if(is.na(HSC_Pchic_file_test_OE[row, "Other_end_ID"])){
    i + row
  }else{
    ID
  }
})
HSC_Pchic_cleaned <- rbind(HSC_Pchic_file_test[!is.na(HSC_Pchic_file_test$Other_end_ID),], HSC_Pchic_file_test_OE)
HSC_Pchic_cleaned <- HSC_Pchic_cleaned[,3:14]
HSC_Pchic_cleaned %>% write.table("../Datasets/HSC_Pchic_cleaned.tsv", sep = "\t", row.names = F)
```


```{r}
HSC_Pchic_cleaned <- read.table("../Datasets/HSC_Pchic_cleaned.tsv", sep = "\t", header = T)
HSC_pchic_bed <- HSC_Pchic_cleaned[,c(1:8, 11, 12)]
colnames(HSC_pchic_bed) <- c(rep(c("chr", "start", "end", "Name"), 2), "ID", "ID")
HSC_pchic_bed <- rbind(HSC_pchic_bed[,c(1:4, 9)], HSC_pchic_bed[,c(5:8,10)]) %>% unique
HSC_Pchic_edges <- HSC_Pchic_cleaned[,c(11,12)]
colnames(HSC_Pchic_edges) <- c("source", "target")
HSC_pchic_Granges <- GRanges(seqnames = HSC_pchic_bed$chr, 
                         ranges = IRanges(start = HSC_pchic_bed$start, end = HSC_pchic_bed$end), 
                         ID = HSC_pchic_bed$ID)
```

```{r}
overlaps <- findOverlaps(dCPGs_Clusters_GRanges, HSC_pchic_Granges)
dCPGs_Clusters_HSCpchic <- data.frame(mcols(dCPGs_Clusters_GRanges[queryHits(overlaps),]),
                          data.frame(mcols(HSC_pchic_Granges[subjectHits(overlaps),]))) %>% unique()

HSC_pchic_net <- dplyr::filter(HSC_Pchic_edges, source %in% dCPGs_Clusters_HSCpchic$ID | target %in% dCPGs_Clusters_HSCpchic$ID)

dCPGs_filtered_Clusters_annotated <- merge(dCPGs_filtered_Clusters_annotated, dCPGs_Clusters_HSCpchic, by.x = "ID", by.y = "CpG")
colnames(dCPGs_filtered_Clusters_annotated)[11] <- "fragment_ID"
write.table(dCPGs_filtered_Clusters_annotated, "../Results/DNAmethylation/dCPGs_Clusters_HSC_pchic.tsv", sep = "\t", row.names = F)
write.table(HSC_pchic_net, "../Results/Chromatine_part/HSC_pchic_dCPGs.tsv", sep = "\t", row.names = F)
```

```{r}
HSC_pchic_gene_network <- HSC_Pchic_file_test[,c(3:6, 13)] %>% unique()
```

```{r}
library('biomaRt')
mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
genes <- HSC_pchic_gene_network$bait_name %>% sapply(function(genes){
  stringr::str_split(genes, pattern = ",")
}) %>% c() %>% unlist() %>% unique
G_list <- getBM(filters= "ensembl_transcript_id", attributes= c("hgnc_symbol", "ensembl_transcript_id"),values=genes,mart= mart)
G_list <- G_list[stringr::str_detect(G_list$hgnc_symbol, pattern = "[:alpha:]"),]
```


```{r}
HSC_pchic_gene_network <- tidyr::separate_rows(HSC_pchic_gene_network, bait_name, sep = ",")
HSC_pchic_gene_network <- unique(HSC_pchic_gene_network)
HSC_pchic_gene_network <- HSC_pchic_gene_network[,c("bait_ID", "bait_name")]
colnames(HSC_pchic_gene_network) <- c("source", "target")
HSC_pchic_gene_network_annotated <- merge(HSC_pchic_gene_network, G_list, by.x = "target", by.y = "ensembl_transcript_id", all.x = T) %>% unique

HSC_Pchic_cleaned_gene_named <- merge(HSC_Pchic_cleaned, HSC_pchic_gene_network_annotated, by.x = "bait_ID", by.y = "source", all.x = T) %>%
  merge(., HSC_pchic_gene_network_annotated, by.x = "Other_end_ID", by.y = "source", all.x = T)

HSC_Pchic_cleaned_gene_named$bait_name <- HSC_Pchic_cleaned_gene_named$hgnc_symbol.x
HSC_Pchic_cleaned_gene_named$otherEnd_name <- HSC_Pchic_cleaned_gene_named$hgnc_symbol.y

HSC_Pchic_cleaned_gene_named <- HSC_Pchic_cleaned_gene_named %>% 
  .[,c(2,1, 3:12)]  %>% 
  unique()

HSC_Pchic_cleaned_gene_named %>%
  write.table("../Datasets/HSC_Pchic_cleaned_gene_named.tsv", sep = "\t", row.names = F)

HSC_pchic_gene_network_annotated <- dplyr::filter(HSC_pchic_gene_network_annotated, hgnc_symbol %ni% c("snoU13", "Y_RNA"))
HSC_pchic_gene_network_annotated <- HSC_pchic_gene_network_annotated[,2:3]
colnames(HSC_pchic_gene_network_annotated) <- c("source", "target")
HSC_pchic_gene_network_annotated <- na.omit(HSC_pchic_gene_network_annotated)
```

```{r}
Interconnect_all_node_from_vector <- function(vec){
  len <- length(vec)
  source <- sapply(1:len, function(element){
    rep(vec[element], len-element)
  }) %>% unlist()
  target <- sapply(1:(len-1), function(element){
    vec[(1+element):len]
  }) %>% unlist()
  data.frame("source" = source,
             "target" = target)
}

Replace_fragments_promoter_to_gene_names <- function(pchic_network, fragment_gene_annotation){
  res <- lapply(1:nrow(pchic_network), function(n_row){
    fragment_A_gene <- dplyr::filter(fragment_gene_annotation, source == pchic_network[n_row, "source"]) %>% .$target
    if (length(fragment_A_gene) == 0){
      fragment_A_gene <- pchic_network[n_row, "source"]
    }
    fragment_B_gene <- dplyr::filter(fragment_gene_annotation, source == pchic_network[n_row, "target"]) %>% .$target
    if (length(fragment_B_gene) == 0){
      fragment_B_gene <- pchic_network[n_row, "target"]
    }
    vec <- unique(c(fragment_A_gene, fragment_B_gene))
    
    if(n_row %% floor(nrow(pchic_network)/100) == 0){
      message(paste0(n_row %/% floor(nrow(pchic_network)/100), "%!"))
    }
    if(length(vec) == 1){
      data.frame("source" = pchic_network[n_row, "source"], "target"= pchic_network[n_row, "target"])
    }else{
      Interconnect_all_node_from_vector(vec)
    }
  })

  res <- rbindlist(res)
  unique(res)
}
```

```{r}
Promoter_fragment <- Replace_fragments_promoter_to_gene_names(HSC_Pchic_edges, HSC_pchic_gene_network_annotated)

```

```{r}
c(Promoter_fragment$source, Promoter_fragment$target) %>% unique %>% length
Promoter_fragment %>% write.table("../Results/Chromatine_part/Promoter_named_HSC_pchic.tsv", sep = "\t", quote = F, row.names = F)
```

```{r}
Fragments_features <- merge(dCPGs_filtered_Clusters_annotated, HSC_pchic_gene_network_annotated, by.x = "fragment_ID", by.y = "source", all.x = T)
Fragments_features$target <- sapply(1:nrow(Fragments_features), function(fragment){
  if(is.na(Fragments_features[fragment,"target"])){
    Fragments_features[fragment,"fragment_ID"]
  }else{
    Fragments_features[fragment, "target"]
  }
})

Fragments_features_promoter_fragments <- dplyr::filter(Fragments_features, target %in% Promoter_fragment$source | target %in% Promoter_fragment$target)

Fragments_features_promoter_fragments <- split(Fragments_features_promoter_fragments, Fragments_features_promoter_fragments$target)

Fragments_features_promoter_fragments <- lapply(names(Fragments_features_promoter_fragments), function(gene){
  data.frame("name" = gene, 
             "logFC" = mean(Fragments_features_promoter_fragments[[gene]]$logFC), 
             "P.Value" = max(Fragments_features_promoter_fragments[[gene]]$P.Value))
}) %>% rbindlist()

write.table(Fragments_features_promoter_fragments, "../Results/Chromatine_part/Features_HSC_fragment_methylation.tsv", sep = "\t", quote = F, row.names = F)
```

