---
title: "Koichi_Cluster_gene_expression_analysis"
author: Alexis Hucteau
date: "`r Sys.Date()`"
output:
    html_document:
      toc: yes
      toc_float: yes
      theme: united
editor_options:
  markdown:
    wrap: sentence
---

# Libraries

```{r, error=F, warning=F, echo=FALSE, include=F}
rm(DmIALite, DOLite, DOLiteTerm, HsIALite, MmIALite, RnIALite)

library(FactoMineR)
library(igraph)
library(limma)
library(viper)
library(stringr)
library(data.table)
library(dplyr)
library(RCy3)
library(factoextra)
library(pheatmap)
library(Hmisc)
library(matrixStats)
library(RColorBrewer)
library(EnhancedVolcano)
library(clusterProfiler)
library(enrichplot)
library(ggnewscale)
library(DOSE)
library(biomaRt)
library(org.Hs.eg.db)
library(aracne.networks)
library(sva)
library(ComplexHeatmap)
library(ggpubr)



"%ni%" <- Negate("%in%")
source("~/Core_scripts/msviper_functions.R")
source("~/Core_scripts/core_functions.R")
source("~/GitHub/Multiplex_DNAmet_PPI_Chrom_Coexp/Make_koichi_factor.R")
```

# Load RNAseq data

```{r}
RNAseq <- read.csv("~/GitHub/Koichi_gene_expression_analyses_git/Koichi_gene_expression_analyses/DATA/RNAseq_parsed.csv", row.names = 1, header = T, check.names = F)

Clinical_patient_data <- read.table("../Datasets/Clinical_Koichi_data.tsv", sep = "\t", header = T)
Phenotype_Cluster <- sapply(colnames(RNAseq), function(sample){
  if(sample %in% Clinical_patient_data$Relapse_RNAseq_data){
    "Relapse"
  }else{
    Clinical_patient_data %>% dplyr::filter(Baseline_RNAseq_data == sample) %>% .$Cluster
  }
})
RNAseq <- RNAseq[,Phenotype_Cluster %in% c("Cluster_NR", "Cluster_R")]
Phenotype_Cluster <- Phenotype_Cluster[Phenotype_Cluster %in% c("Cluster_NR", "Cluster_R")]
write.table(RNAseq, "../Datasets/RNAseq_Baseline.tsv", sep = "\t", row.names = F)
```

```{r}
Pheno_mIDH <- sapply(colnames(RNAseq), function(sample){
  dplyr::filter(Clinical_patient_data, Baseline_RNAseq_data == sample) %>% .$IDH_isoform
})

Phenotype_Cluster_mIDH <- paste(Pheno_mIDH, Phenotype_Cluster, sep = "_")
```


# PCA

```{r}
res.pca <- prcomp(t(RNAseq))
p <- fviz_pca_ind(res.pca, label="all", habillage=Phenotype_Cluster_mIDH,
                               addEllipses=T, ellipse.level=0.95)
p <- p + ggtitle("PCA gene expression")
p
png("../Results/DGEA/PCA_Koichi_data_Clusters_mIDH.png")
p
dev.off()
```

```{r}
Make_heatmap <- function(DATA, Phenotype, method = "pearson", title, annotation_color, kmeans_k = NA, cuttree = NA, corr=T, TF=F) {
  annotation_for_heatmap <- data.frame(Phenotype = Phenotype$Phenotype)
  rownames(annotation_for_heatmap) <- colnames(DATA)
  if(corr){
    corr <- rcorr(as.matrix(DATA), type = method)$r
    colnames(corr) <- colnames(DATA)
    if(TF){
      color= rev(colorRampPalette(brewer.pal(n = 9, name = "RdBu"))(100))
    }else{
      color = colorRampPalette(brewer.pal(n = 9, name = "YlOrRd"))(100)
    }
  }else{
    corr=DATA
    color= rev(colorRampPalette(brewer.pal(n = 9, name = "RdBu"))(100))
  }
  
  title <- paste0(title, " ", method)
  heatmap <- pheatmap(corr, 
                      color = color,
                      annotation_col = annotation_for_heatmap,
                      annotation_colors = annotation_color,
                      legend = TRUE,
                      treeheight_row = 20,
                      main = title, 
                      fontsize = 10,
                      cutree_cols = cuttree
  )
  return(heatmap)
}
```

```{r}
Var_gene <- rowVars(as.matrix(RNAseq))
names(Var_gene) <- rownames(RNAseq)
```

```{r}
nb_genes <- 100
method <- "pearson"
Top_var_genes <- Var_gene[order(Var_gene, decreasing = T)] %>% .[1:nb_genes]
Top_Var_RNAseq <- RNAseq %>% .[rownames(.) %in% names(Top_var_genes),] 
ann_color_RNA <- list(
    Phenotype = c(Cluster_R = "blue", Cluster_NR = "red"))

pheno_heatmap <- data.frame("Responder" = Phenotype_Cluster == "Cluster_R", "Non_responder" = Phenotype_Cluster == "Cluster_NR")
pheno_heatmap$Phenotype <- Phenotype_Cluster
heat <- Make_heatmap(Top_Var_RNAseq, Phenotype = pheno_heatmap, title = paste0("Patient correlation of Top ", nb_genes, " variable Gene expression"), annotation_color = ann_color_RNA, method = method)

png(paste0("../Results/DGEA/Heatmap_Clusters_GE_top_", nb_genes,"_corr_", method, "_Koichi_data.png"))
heat
dev.off()

res.pca <- prcomp(t(Top_Var_RNAseq))
p <- fviz_pca_ind(res.pca, label="all", habillage=Phenotype_Cluster,
                               addEllipses=T, ellipse.level=0.95)
p <- p + ggtitle(paste0("PCA top ", nb_genes, " genes"))
p
png(paste0("../Results/DGEA/PCA_Clusters_GE_top_", nb_genes,"_Koichi_data.png"))
p
dev.off()
if(nb_genes < 500){
  heat2 <- Make_heatmap(Top_Var_RNAseq, Phenotype = pheno_heatmap, title = paste0("Patient Clusters correlation of Top ", nb_genes, " variable Gene expression"), annotation_color = ann_color_RNA, method = "", corr = F)
  png(paste0("../Results/DGEA/Heatmap_Clusters_GE_top_", nb_genes,"_Koichi_data.png"))
  heat2
  dev.off()
}
```

```{r}
data(regulonlaml)
viper_regulons2dorothea <- function(r) {
  res <- r %>%
    purrr::map_df(
      .f = function(i) {
        tf_target <- i$tfmode %>%
          tibble::enframe(name = "target", value = "mor") %>%
          mutate(likelihood = i$likelihood)
      },
      .id = "tf"
    )
  return(res)
}

regulonaml <- viper_regulons2dorothea(regulonlaml)
regulonaml_SYMBOL <- data.frame("tf" = GeneAnswers::getSymbols(regulonaml$tf, data = "org.Hs.eg"),
                                "target" = GeneAnswers::getSymbols(regulonaml$target, data = "org.Hs.eg"),
                                "mor" = regulonaml$mor,
                                "likelihood" = regulonaml$likelihood)
regulonaml_SYMBOL <- dplyr::filter(regulonaml_SYMBOL, likelihood > 0.85550)
rm(regulonlaml)
data(dorothea_hs, package = "dorothea")
dorothea_regulon <- dorothea_hs %>%
  filter(confidence %in% c("A", "B"))
dorothea_regulon <- dorothea_regulon[,c(1,3)]
dorothea_regulon_D <- dorothea_hs %>%
  filter(confidence %in% c("A", "B", "C", "D"))
dorothea_regulon_D <- dorothea_regulon_D[,c(1,3)]

All_sample_ARACNe_network <- read.table("../../Multiplex_DNAmet_PPI_Chrom_Coexp/Output_all_Samples_TPM/network.txt", sep = "\t", header = T)
colnames(All_sample_ARACNe_network)[1:2] <- c("tf", "target")

ref_Cluster1 <- Phenotype_Cluster == "Cluster_R"
ref_Cluster2 <- Phenotype_Cluster == "Cluster_NR"

ref_Cluster1_mIDH2 <- Phenotype_Cluster_mIDH == "IDH2_Cluster_R"
ref_Cluster1_mIDH1 <- Phenotype_Cluster_mIDH == "IDH1_Cluster_R"
ref_Cluster2_mIDH2 <- Phenotype_Cluster_mIDH == "IDH2_Cluster_NR"
ref_Cluster2_mIDH1 <- Phenotype_Cluster_mIDH == "IDH1_Cluster_NR"
```

# msVIPER TF enrichment analysis

## Cluster analysis

```{r}
Clusters_msvip_ARACNe <- run_msviper(RNAseq, All_sample_ARACNe_network, use_aracne = T, ref_Cluster1, ref_Cluster2, "Cluster1", "Cluster2", minsize = 1, ges.filter = T)
Clusters_msvip_Dorothea <- run_msviper(RNAseq, dorothea_regulon, use_aracne = T, ref_Cluster1, ref_Cluster2, "Cluster1", "Cluster2", minsize = 1, ges.filter = T)
Clusters_msvip_Dorothea_D <- run_msviper(RNAseq, dorothea_regulon_D, use_aracne = T, ref_Cluster1, ref_Cluster2, "Cluster1", "Cluster2", minsize = 1, ges.filter = T)
Clusters_msvip_Regulonlaml <- run_msviper(RNAseq, regulonaml_SYMBOL, use_aracne = T, ref_Cluster1, ref_Cluster2, "Cluster1", "Cluster2", minsize = 1, ges.filter = T)

Koichi_TF_actitity_ARACNe <- viper(RNAseq, dorothea2viper_regulons(Clusters_msvip_ARACNe$regulons))
Koichi_TF_actitity_Dorothea <- viper(RNAseq, dorothea2viper_regulons(Clusters_msvip_Dorothea$regulons))
Koichi_TF_actitity_Regulonlaml <- viper(RNAseq, dorothea2viper_regulons(Clusters_msvip_Regulonlaml$regulons))
Koichi_TF_actitity_Dorothea_D <- viper(RNAseq, dorothea2viper_regulons(Clusters_msvip_Dorothea_D$regulons))

```

## Cluster & IDH2 analysis

```{r}
R2_vs_NR2_msvip_ARACNe <- run_msviper(RNAseq, All_sample_ARACNe_network, use_aracne = T, ref_Cluster1_mIDH2, ref_Cluster2_mIDH2, "Cluster1_mIDH2", "Cluster2_mIDH2", minsize = 1, ges.filter = T)
R2_vs_NR2_msvip_Dorothea <- run_msviper(RNAseq, dorothea_regulon, use_aracne = T, ref_Cluster1_mIDH2, ref_Cluster2_mIDH2, "Cluster1_mIDH2", "Cluster2_mIDH2", minsize = 1, ges.filter = T)
R2_vs_NR2_msvip_Dorothea_D <- run_msviper(RNAseq, dorothea_regulon_D, use_aracne = T, ref_Cluster1_mIDH2, ref_Cluster2_mIDH2, "Cluster1_mIDH2", "Cluster2_mIDH2", minsize = 1, ges.filter = T)
R2_vs_NR2_msvip_Regulonlaml <- run_msviper(RNAseq, regulonaml_SYMBOL, use_aracne = T, ref_Cluster1_mIDH2, ref_Cluster2_mIDH2, "Cluster1_mIDH2", "Cluster2_mIDH2", minsize = 1, ges.filter = T)
```

## Cluster R IDH1 vs Cluster NR IDH2 analysis

```{r}
R1_vs_NR2_msvip_ARACNe <- run_msviper(RNAseq, All_sample_ARACNe_network, use_aracne = T, ref_Cluster1_mIDH1, ref_Cluster2_mIDH2, "Cluster1_mIDH1", "Cluster2_mIDH2", minsize = 1, ges.filter = T)
R1_vs_NR2_msvip_Dorothea <- run_msviper(RNAseq, dorothea_regulon, use_aracne = T, ref_Cluster1_mIDH1, ref_Cluster2_mIDH2, "Cluster1_mIDH1", "Cluster2_mIDH2", minsize = 1, ges.filter = T)
R1_vs_NR2_msvip_Dorothea_D <- run_msviper(RNAseq, dorothea_regulon_D, use_aracne = T, ref_Cluster1_mIDH1, ref_Cluster2_mIDH2, "Cluster1_mIDH1", "Cluster2_mIDH2", minsize = 1, ges.filter = T)
R1_vs_NR2_msvip_Regulonlaml <- run_msviper(RNAseq, regulonaml_SYMBOL, use_aracne = T, ref_Cluster1_mIDH1, ref_Cluster2_mIDH2, "Cluster1_mIDH1", "Cluster2_mIDH2", minsize = 1, ges.filter = T)
```

## Plots Clusters 

### ARACNe

```{r}
png("../Results/DGEA_Clusters_Clean/ARACNe/Cluster_msviplot_ARACNe.png", width = 1280, height = 720)
plot(Clusters_msvip_ARACNe$mrs, mrs = 50)
dev.off()
plot(Clusters_msvip_ARACNe$mrs, mrs = 50)
Clusters_msvip_ARACNe$mrs_table %>% 
  write.table("../Results/DGEA_Clusters_Clean/ARACNe/TF_Diff_activity_Clusters_ARACNe.tsv", sep="\t", row.names = F)
Clusters_msvip_ARACNe$mrs_table %>% dplyr::filter(pval < 0.1) %>% 
  write.table("../Results/DGEA_Clusters_Clean/ARACNe/Filtered_TF_Diff_activity_Clusters_ARACNe.tsv", sep="\t", row.names = F)
Clusters_msvip_ARACNe$regulons %>% 
  write.table("../Results/DGEA_Clusters_Clean/ARACNe/Cluster_GRN_ARACNe.tsv", sep = "\t", row.names = F)
```

### Dorothea

```{r}
png("../Results/DGEA_Clusters_Clean/Dorothea/Cluster_msviplot_Dorothea.png", width = 1280, height = 720)
plot(Clusters_msvip_Dorothea$mrs, mrs = 50)
dev.off()
plot(Clusters_msvip_Dorothea$mrs, mrs = 50)
Clusters_msvip_Dorothea$mrs_table %>% 
  write.table("../Results/DGEA_Clusters_Clean/Dorothea/TF_Diff_activity_Clusters_Dorothea.tsv", sep="\t", row.names = F)
Clusters_msvip_Dorothea$mrs_table %>% dplyr::filter(pval < 0.1) %>% 
  write.table("../Results/DGEA_Clusters_Clean/Dorothea/Filtered_TF_Diff_activity_Clusters_Dorothea.tsv", sep="\t", row.names = F)
Clusters_msvip_Dorothea$regulons %>% 
  write.table("../Results/DGEA_Clusters_Clean/Dorothea/Cluster_GRN_Dorothea.tsv", sep = "\t", row.names = F)
```

### Dorothea D

```{r}
png("../Results/DGEA_Clusters_Clean/Dorothea_D/Cluster_msviplot_Dorothea_D.png", width = 1280, height = 720)
plot(Clusters_msvip_Dorothea_D$mrs, mrs = 50)
dev.off()
plot(Clusters_msvip_Dorothea_D$mrs, mrs = 50)
Clusters_msvip_Dorothea_D$mrs_table %>% 
  write.table("../Results/DGEA_Clusters_Clean/Dorothea_D/TF_Diff_activity_Clusters_Dorothea_D.tsv", sep="\t", row.names = F)
Clusters_msvip_Dorothea_D$mrs_table %>% dplyr::filter(pval < 0.1) %>% 
  write.table("../Results/DGEA_Clusters_Clean/Dorothea_D/Filtered_TF_Diff_activity_Clusters_Dorothea_D.tsv", sep="\t", row.names = F)
Clusters_msvip_Dorothea_D$regulons %>% 
  write.table("../Results/DGEA_Clusters_Clean/Dorothea_D/Cluster_GRN_Dorothea.tsv", sep = "\t", row.names = F)
```

### RegulonlAML

```{r}
png("../Results/DGEA_Clusters_Clean/Regulonlaml/Cluster_msviplot_Regulonlaml.png", width = 1280, height = 720)
plot(Clusters_msvip_Regulonlaml$mrs, mrs = 50)
dev.off()
plot(Clusters_msvip_Regulonlaml$mrs, mrs = 50)
Clusters_msvip_Regulonlaml$mrs_table %>% 
  write.table("../Results/DGEA_Clusters_Clean/Regulonlaml/TF_Diff_activity_Clusters_Regulonlaml.tsv", sep="\t", row.names = F)
Clusters_msvip_Regulonlaml$mrs_table %>% dplyr::filter(pval < 0.1) %>% 
  write.table("../Results/DGEA_Clusters_Clean/Regulonlaml/Filtered_TF_Diff_activity_Clusters_Regulonlaml.tsv", sep="\t", row.names = F)
Clusters_msvip_Regulonlaml$regulons %>% 
  write.table("../Results/DGEA_Clusters_Clean/Regulonlaml/Cluster_GRN_Regulonlaml.tsv", sep = "\t", row.names = F)
```



## Plots Clusters & mIDH2

### ARACNe

```{r}
png("../Results/Cluster_IDH2/ARACNe/R2_vs_NR2_msvip_ARACNe.png", width = 1280, height = 720)
plot(R2_vs_NR2_msvip_ARACNe$mrs, mrs = 50)
dev.off()
plot(R2_vs_NR2_msvip_ARACNe$mrs, mrs = 50)
R2_vs_NR2_msvip_ARACNe$mrs_table %>% 
  write.table("../Results/Cluster_IDH2/ARACNe/TF_Diff_activity_R2_vs_NR2_msvip_ARACNe.tsv", sep="\t", row.names = F)
R2_vs_NR2_msvip_ARACNe$mrs_table %>% dplyr::filter(pval < 0.1) %>% 
  write.table("../Results/Cluster_IDH2/ARACNe/Filtered_TF_Diff_activity_R2_vs_NR2_msvip_ARACNe.tsv", sep="\t", row.names = F)
R2_vs_NR2_msvip_ARACNe$regulons %>% 
  write.table("../Results/Cluster_IDH2/ARACNe/R2_vs_NR2_GRN_ARACNe.tsv", sep = "\t", row.names = F)
```

### Dorothea

```{r}
png("../Results/Cluster_IDH2/Dorothea/R2_vs_NR2_msvip_Dorothea.png", width = 1280, height = 720)
plot(R2_vs_NR2_msvip_Dorothea$mrs, mrs = 50)
dev.off()
plot(R2_vs_NR2_msvip_Dorothea$mrs, mrs = 50)
R2_vs_NR2_msvip_Dorothea$mrs_table %>% 
  write.table("../Results/Cluster_IDH2/Dorothea/TF_Diff_activity_R2_vs_NR2_msvip_Dorothea.tsv", sep="\t", row.names = F)
R2_vs_NR2_msvip_Dorothea$mrs_table %>% dplyr::filter(pval < 0.1) %>% 
  write.table("../Results/Cluster_IDH2/Dorothea/Filtered_TF_Diff_activity_R2_vs_NR2_msvip_Dorothea.tsv", sep="\t", row.names = F)
R2_vs_NR2_msvip_Dorothea$regulons %>% 
  write.table("../Results/Cluster_IDH2/Dorothea/R2_vs_NR2_GRN_Dorothea.tsv", sep = "\t", row.names = F)
```

### Dorothea D

```{r}
png("../Results/Cluster_IDH2/Dorothea_D/R2_vs_NR2_msvip_Dorothea_D.png", width = 1280, height = 720)
plot(R2_vs_NR2_msvip_Dorothea_D$mrs, mrs = 50)
dev.off()
plot(R2_vs_NR2_msvip_Dorothea_D$mrs, mrs = 50)
R2_vs_NR2_msvip_Dorothea_D$mrs_table %>% 
  write.table("../Results/Cluster_IDH2/Dorothea_D/TF_Diff_activity_R2_vs_NR2_msvip_Dorothea_D.tsv", sep="\t", row.names = F)
R2_vs_NR2_msvip_Dorothea_D$mrs_table %>% dplyr::filter(pval < 0.1) %>% 
  write.table("../Results/Cluster_IDH2/Dorothea_D/Filtered_TF_Diff_activity_R2_vs_NR2_msvip_Dorothea_D.tsv", sep="\t", row.names = F)
R2_vs_NR2_msvip_Dorothea_D$regulons %>% 
  write.table("../Results/Cluster_IDH2/Dorothea_D/R2_vs_NR2_GRN_Dorothea_D.tsv", sep = "\t", row.names = F)
```

### RegulonlAML

```{r}
png("../Results/Cluster_IDH2/Regulonlaml/R2_vs_NR2_msvip_Regulonlaml.png", width = 1280, height = 720)
plot(R2_vs_NR2_msvip_Regulonlaml$mrs, mrs = 50)
dev.off()
plot(R2_vs_NR2_msvip_Regulonlaml$mrs, mrs = 50)
R2_vs_NR2_msvip_Regulonlaml$mrs_table %>% 
  write.table("../Results/Cluster_IDH2/Regulonlaml/TF_Diff_activity_R2_vs_NR2_msvip_Regulonlaml.tsv", sep="\t", row.names = F)
R2_vs_NR2_msvip_Regulonlaml$mrs_table %>% dplyr::filter(pval < 0.1) %>% 
  write.table("../Results/Cluster_IDH2/Regulonlaml/Filtered_TF_Diff_activity_R2_vs_NR2_msvip_Regulonlaml.tsv", sep="\t", row.names = F)
R2_vs_NR2_msvip_Regulonlaml$regulons %>% 
  write.table("../Results/Cluster_IDH2/Regulonlaml/R2_vs_NR2_GRN_Regulonlaml.tsv", sep = "\t", row.names = F)
```

## Plots Cluster NR IDH2 & R IDH1

### ARACNe

```{r}
png("../Results/NR_IDH2_R_IDH1/ARACNe/R1_vs_NR2_msvip_ARACNe.png", width = 1280, height = 720)
plot(R1_vs_NR2_msvip_ARACNe$mrs, mrs = 50)
dev.off()
plot(R1_vs_NR2_msvip_ARACNe$mrs, mrs = 50)
R1_vs_NR2_msvip_ARACNe$mrs_table %>% 
  write.table("../Results/NR_IDH2_R_IDH1/ARACNe/TF_Diff_activity_R1_vs_NR2_msvip_ARACNe.tsv", sep="\t", row.names = F)
R1_vs_NR2_msvip_ARACNe$mrs_table %>% dplyr::filter(pval < 0.1) %>% 
  write.table("../Results/NR_IDH2_R_IDH1/ARACNe/Filtered_TF_Diff_activity_R1_vs_NR2_msvip_ARACNe.tsv", sep="\t", row.names = F)
R1_vs_NR2_msvip_ARACNe$regulons %>% 
  write.table("../Results/NR_IDH2_R_IDH1/ARACNe/R1_vs_NR2_GRN_ARACNe.tsv", sep = "\t", row.names = F)
```

### Dorothea

```{r}
png("../Results/NR_IDH2_R_IDH1/Dorothea/R1_vs_NR2_msvip_Dorothea.png", width = 1280, height = 720)
plot(R1_vs_NR2_msvip_Dorothea$mrs, mrs = 50)
dev.off()
plot(R1_vs_NR2_msvip_Dorothea$mrs, mrs = 50)
R1_vs_NR2_msvip_Dorothea$mrs_table %>% 
  write.table("../Results/NR_IDH2_R_IDH1/Dorothea/TF_Diff_activity_R1_vs_NR2_msvip_Dorothea.tsv", sep="\t", row.names = F)
R1_vs_NR2_msvip_Dorothea$mrs_table %>% dplyr::filter(pval < 0.1) %>% 
  write.table("../Results/NR_IDH2_R_IDH1/Dorothea/Filtered_TF_Diff_activity_R1_vs_NR2_msvip_Dorothea.tsv", sep="\t", row.names = F)
R1_vs_NR2_msvip_Dorothea$regulons %>% 
  write.table("../Results/NR_IDH2_R_IDH1/Dorothea/R1_vs_NR2_GRN_Dorothea.tsv", sep = "\t", row.names = F)
```

### Dorothea D

```{r}
png("../Results/NR_IDH2_R_IDH1/Dorothea_D/R1_vs_NR2_msvip_Dorothea_D.png", width = 1280, height = 720)
plot(R1_vs_NR2_msvip_Dorothea_D$mrs, mrs = 50)
dev.off()
plot(R1_vs_NR2_msvip_Dorothea_D$mrs, mrs = 50)
R1_vs_NR2_msvip_Dorothea_D$mrs_table %>% 
  write.table("../Results/NR_IDH2_R_IDH1/Dorothea_D/TF_Diff_activity_R1_vs_NR2_msvip_Dorothea_D.tsv", sep="\t", row.names = F)
R1_vs_NR2_msvip_Dorothea_D$mrs_table %>% dplyr::filter(pval < 0.1) %>% 
  write.table("../Results/NR_IDH2_R_IDH1/Dorothea/Filtered_TF_Diff_activity_R1_vs_NR2_msvip_Dorothea_D.tsv", sep="\t", row.names = F)
R1_vs_NR2_msvip_Dorothea_D$regulons %>% 
  write.table("../Results/NR_IDH2_R_IDH1/Dorothea_D/R1_vs_NR2_GRN_Dorothea_D.tsv", sep = "\t", row.names = F)
```

### RegulonlAML

```{r}
png("../Results/NR_IDH2_R_IDH1/Regulonlaml/R1_vs_NR2_msvip_Regulonlaml.png", width = 1280, height = 720)
plot(R1_vs_NR2_msvip_Regulonlaml$mrs, mrs = 50)
dev.off()
plot(R1_vs_NR2_msvip_Regulonlaml$mrs, mrs = 50)
R1_vs_NR2_msvip_Regulonlaml$mrs_table %>% 
  write.table("../Results/NR_IDH2_R_IDH1/Regulonlaml/TF_Diff_activity_R1_vs_NR2_msvip_Regulonlaml.tsv", sep="\t", row.names = F)
R1_vs_NR2_msvip_Regulonlaml$mrs_table %>% dplyr::filter(pval < 0.1) %>% 
  write.table("../Results/NR_IDH2_R_IDH1/Regulonlaml/Filtered_TF_Diff_activity_R1_vs_NR2_msvip_Regulonlaml.tsv", sep="\t", row.names = F)
R1_vs_NR2_msvip_Regulonlaml$regulons %>% 
  write.table("../Results/NR_IDH2_R_IDH1/Regulonlaml/R1_vs_NR2_GRN_Regulonlaml.tsv", sep = "\t", row.names = F)
```

# Overlaps TF enrichments

## Functions

```{r}
Do_overlaps_TF_upsets <- function(List_of_samples, title){
  L <- lapply(List_of_samples, function(TF){
    dplyr::filter(TF, pval < 0.05) %>% .$TF
  })
  
  comb_mat <- ComplexHeatmap::make_comb_mat(L)
  comb_mat <- comb_mat[colSums(comb_mat) >1]
  cs <- comb_size(comb_mat)

  ht <- ComplexHeatmap::UpSet(comb_mat, row_title = title, top_annotation = HeatmapAnnotation(
        "Intersections" = anno_barplot(cs, 
            ylim = c(0, max(cs)*1.1),
            border = FALSE, 
            gp = gpar(fill = "black"), 
            height = unit(4, "cm")
        ), 
        annotation_name_side = "left", 
        annotation_name_rot = 90))
  ht = draw(ht)
  od = column_order(ht)
  decorate_annotation("Intersections", {
      grid.text(cs[od], x = seq_along(cs), y = unit(cs[od], "native") + unit(2, "pt"), 
          default.units = "native", just = c("left", "bottom"), 
          gp = gpar(fontsize = 12, col = "#404040"), rot = 45)
  })
}
```

## Analysis

```{r}
p <- Do_overlaps_TF_upsets(List_of_samples = list(Regulonlaml = Clusters_msvip_Regulonlaml$mrs_table, ARACNe = Clusters_msvip_ARACNe$mrs_table, Dorothea = Clusters_msvip_Dorothea$mrs_table), title = "Clusters_TF_intersections")
p
png("../Results/DGEA_Clusters_Clean/Diff_TF_upsetplots.png")
p
dev.off()
```

# TF analysis

```{r}
Var_TF <- rowVars(as.matrix(Koichi_TF_actitity_ARACNe))
names(Var_TF) <- rownames(Koichi_TF_actitity_ARACNe)
```

```{r}
nb_TF <- 5
method <- "pearson"
Top_var_TF <- Var_TF[order(Var_TF, decreasing = T)] %>% .[1:nb_TF]
Top_Var_TF <- Koichi_TF_actitity_ARACNe %>% .[rownames(.) %in% names(Top_var_TF),] 
heat <- Make_heatmap(Top_Var_TF, Phenotype = pheno_heatmap, title = paste0("Patient Clusters correlation of Top ", nb_TF, " variable TF"), annotation_color = ann_color_RNA, method = method,TF=T)

png(paste0("../Results/DGEA_Clusters_Clean/ARACNe/Heatmap/Heatmap_Clusters_TF_top_", nb_TF,"_corr_", method, "_Koichi_data.png"))
heat
dev.off()
res.pca <- Top_Var_TF %>% as.data.frame %>% t() %>% prcomp
p <- fviz_pca_ind(res.pca, label="all", habillage=Phenotype_Cluster,
                               addEllipses=T, ellipse.level=0.95)
p <- p + ggtitle(paste0("PCA Clusters top ", nb_TF, " TF"))
p
png(paste0("../Results/DGEA_Clusters_Clean/ARACNe/PCA/PCA_Clusters_GE_top_", nb_TF,"_TF_Koichi_data.png"))
p
dev.off()
heat2 <- Make_heatmap(Top_Var_TF, Phenotype = pheno_heatmap, title = paste0("Patient Clusters correlation of Top ", nb_TF, " variable TF"), annotation_color = ann_color_RNA, method = "", corr=F)
png(paste0("../Results/DGEA_Clusters_Clean/ARACNe/Heatmap/Heatmap_Clusters_TF_top_", nb_TF,"_Koichi_data.png"))
heat2
dev.off()
```

```{r}
res.pca <- Koichi_TF_actitity_ARACNe %>% as.data.frame %>% prcomp

p <- fviz_pca_ind(res.pca, label="all",
                               addEllipses=F, ellipse.level=0.95)
p <- p + ggtitle("PCA TF activity")
p
png("../Results/DGEA_Clusters_Clean/ARACNe/Cluster_PCA_TF_Koichi_data.png")
p
dev.off()
```

# DGEA

```{r}
Cluster_DEG_analysis <- Differential_analysis(pheno_heatmap$Phenotype, RNAseq)$`Cluster_NR-Cluster_R`
Cluster_DEG_analysis %>% dplyr::filter(abs(logFC) > 1.5 & P.Value < 0.1) %>%
  write.table("../Results/DGEA/DEGs_tables/Koichi_Cluster_filtered_DEG_analysis.tsv", sep="\t", col.names = NA)
```

## Volcano plot

```{r}
Volcano <- EnhancedVolcano(Cluster_DEG_analysis, 
                           lab = rownames(Cluster_DEG_analysis), 
                           x = 'logFC', y = 'P.Value', 
                           FCcutoff = 0.75, pCutoff = 0.05,
                           title="Cluster DGE", 
                           shape = 1)
Volcano
png("../Results/DGEA/Volcano_plots/Cluster/Clusters_DEG_volcanoplot.png")
Volcano
dev.off()
```

# Overlap DGEA & Enrichments

```{r}
Gene_Up <- Cluster_DEG_analysis %>% dplyr::filter(P.Value < 0.05 & logFC > 1.5) %>% .$ID
Gene_Down <- Cluster_DEG_analysis %>% dplyr::filter(P.Value < 0.05 & logFC < - 1.5) %>% .$ID
GO_UP_enrich_cluster <- enrichGO(gene = Gene_Up, OrgDb = "org.Hs.eg.db", keyType = "SYMBOL")
Go_Down_enrich_cluster <- enrichGO(gene = Gene_Down, OrgDb = "org.Hs.eg.db", keyType = "SYMBOL")

DEGenes <- Cluster_DEG_analysis %>% dplyr::filter(P.Value < 0.05 & abs(logFC) > 1.5) %>% .$logFC
names(DEGenes) <- Cluster_DEG_analysis %>% dplyr::filter(P.Value < 0.05 & abs(logFC) > 1.5) %>% .$ID

DEGenes = sort(DEGenes, decreasing = TRUE)
gse <- gseGO(geneList=DEGenes, 
             ont ="ALL", 
             keyType = "SYMBOL", 
             nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = "org.Hs.eg.db", 
             pAdjustMethod = "none")


png("../Results/DGEA/Enrichments/Cluster/Clusters_GO.png")
dotplot(gse, title = "GO enrichment",showCategory=20, split=".sign") + facet_grid(.~.sign)
dev.off()
dotplot(gse, title = "GO enrichment",showCategory=20, split=".sign") + facet_grid(.~.sign)

hs <- org.Hs.eg.db
DEGenes_ID <- select(hs, names(DEGenes), columns = c("ENTREZID", "SYMBOL"), keytype = "SYMBOL") %>% na.omit() %>% .[!duplicated(.$SYMBOL),]
DEGenes_ID_list <- DEGenes[which(names(DEGenes) %in% DEGenes_ID$SYMBOL)]
names(DEGenes_ID_list) <- DEGenes_ID$ENTREZID

dse <- gseDO(DEGenes_ID_list, pAdjustMethod = "none", pvalueCutoff = 0.1, minGSSize = 5)
ridgeplot(dse)
dotplot(dse, title = "DO enrichment", showCategory=20, split=".sign") + facet_grid(.~.sign)
png("../Results/DGEA/Enrichments/Cluster/Ridgeplot_DO.png")
ridgeplot(dse)
dev.off()
png("../Results/DGEA/Enrichments/Cluster/Dot_DO.png")
dotplot(dse, title = "DO enrichment", showCategory=20, split=".sign") + facet_grid(.~.sign)
dev.off()
kse <- gseKEGG(DEGenes_ID_list, minGSSize = 5, pvalueCutoff = 0.1, pAdjustMethod = "none")
dotplot(kse,showCategory=20, title = "KEGG enrichment",  split=".sign") + facet_grid(.~.sign)
png("../Results/DGEA/Enrichments/Cluster/Dot_KEGG.png")
dotplot(kse, showCategory=20, title = "KEGG enrichment", split=".sign") + facet_grid(.~.sign)
dev.off()
```

```{r}
gse@result %>% write.table("../Results/DGEA/Enrichments/Cluster/gse_Cluster.tsv", sep = "\t", col.names = NA)
kse@result %>% write.table("../Results/DGEA/Enrichments/Cluster/kse_Cluster.tsv", sep = "\t", col.names = NA)
dse@result %>% write.table("../Results/DGEA/Enrichments/Cluster/dse_Cluster.tsv", sep = "\t", col.names = NA)
```

```{r}
enrichGO_UP <- enrichGO(Gene_Up, OrgDb = "org.Hs.eg.db", keyType = "SYMBOL", pvalueCutoff = 0.05, pAdjustMethod = "none", minGSSize = 5)
dotplot(enrichGO_UP,showCategory=15, title = "GO genes up enrichment")
enrichGO_down <- enrichGO(Gene_Down, OrgDb = "org.Hs.eg.db", keyType = "SYMBOL", pvalueCutoff = 0.05, pAdjustMethod = "none", minGSSize = 5)
dotplot(enrichGO_down,showCategory=15, title = "GO genes down enrichment")
goplot(enrichGO_UP)
png("../Results/DGEA/Enrichments/Cluster/Goplot_Up.png")
goplot(enrichGO_UP)
dev.off()
goplot(enrichGO_down)
png("../Results/DGEA/Enrichments/Cluster/Goplot_Down.png")
goplot(enrichGO_down)
dev.off()
kse_readable <- setReadable(kse, 'org.Hs.eg.db', "ENTREZID")
dse_readable <- setReadable(dse, 'org.Hs.eg.db', "ENTREZID")
p1 <- cnetplot(kse_readable, showCategory = 20, foldChange = DEGenes_ID_list, colorEdge=T)

p1 <- cnetplot(kse_readable, node_label="category", showCategory = 20, 
        cex_label_category = 1.2, foldChange = DEGenes_ID_list, colorEdge=T) 
p2 <- cnetplot(kse_readable, node_label="gene", showCategory = 20, 
        cex_label_gene = 0.8, colorEdge=T) 
p3 <- cnetplot(kse_readable, node_label="none", showCategory = 20, 
        color_category='firebrick', 
        color_gene='steelblue', colorEdge=T) 
png("../Results/DGEA/Enrichments/Cluster/KSE_cowplot.png", height = 1080, width = 1920)
cowplot::plot_grid(p1, p2, p3, ncol=2, labels=LETTERS[1:3])
dev.off()
p1 <- cnetplot(kse_readable, node_label="all", showCategory = 20, foldChange = DEGenes_ID_list, color_category='firebrick', colorEdge=T) 
p2 <- cnetplot(dse_readable, node_label="all", showCategory = 20, foldChange = DEGenes_ID_list, color_category='firebrick', colorEdge=T) 
p3 <- cnetplot(gse, node_label="all", showCategory = 20, foldChange = DEGenes_ID_list, color_category='firebrick', colorEdge=T) 

png("../Results/DGEA/Enrichments/Cluster/Cluster_cowplot.png", height = 1080, width = 1920)
cowplot::plot_grid(p1, p2, p3, ncol=2, labels=c("KEGG", "Disease Ontology", "Gene Ontology"))
dev.off()
p2 <- heatplot(dse_readable, foldChange=DEGenes_ID_list, showCategory=10)
p2
edox2 <- pairwise_termsim(dse_readable, showCategory = 200)
p1 <- treeplot(edox2)
p2 <- treeplot(edox2, hclust_method = "average")
aplot::plot_list(p1, p2, tag_levels='A')
png("../Results/DGEA/Enrichments/Cluster/DSE_treeplot.png", height = 1080, width = 1920)
aplot::plot_list(p1, p2, tag_levels='A')
dev.off()
```

# Compare TF enrichments

## Functions

```{r}
Make_TF_nes_enrichment_plot <- function(Tf_feature_1, Tf_feature_2, title = "mIDH_Koichi_TF_enrichment overlap"){
  A <- deparse(substitute(Tf_feature_1)) %>% stringr::str_remove("msvip_")
  B <- deparse(substitute(Tf_feature_2)) %>% stringr::str_remove("msvip_")
  colnames(Tf_feature_1)[1] <- "ID"
  colnames(Tf_feature_2)[1] <- "ID"
  
  TF_1_filtered <- Tf_feature_1 %>%
    dplyr::filter(nes != 0) %>%
    dplyr::select(ID, nes)
  colnames(TF_1_filtered)[2] <- "nes_1"
  TF_2_filtered <- Tf_feature_2 %>%
    dplyr::filter(nes != 0) %>%
    dplyr::select(ID, nes)
  colnames(TF_2_filtered)[2] <- "nes_2"
  TF_comb <- merge(TF_1_filtered, TF_2_filtered, by = "ID", all.x = T, all.y = T)
  p <- ggscatter(TF_comb, x = "nes_1", y = "nes_2", 
                 add = "reg.line", label = "ID", repel = T, size = 0.5,
                 font.label = c(6, "plain"))+
    stat_cor(label.y = 2) +
    stat_regline_equation(label.y = 1.5) +
    labs(main = title, x = A,
         y = B) +
    geom_hline(yintercept = 1, linetype = "dashed") +
    geom_hline(yintercept = -1, linetype = "dashed") +
    geom_vline(xintercept = 1, linetype = "dashed") +
    geom_vline(xintercept = -1, linetype = "dashed") 
  png_path <- paste0("../Results/DGEA_Clusters_Clean/TF_nes_enrichment_plot/", title, "_", A, "_", B , "_Network_enrichment_plot.png")
  print(png_path)
  ggsave(png_path, p)
  p
}
```

## Analysis R2 vs NR2

```{r}
Make_TF_nes_enrichment_plot(R2_vs_NR2_msvip_ARACNe$mrs_table, R2_vs_NR2_msvip_Dorothea$mrs_table)
Make_TF_nes_enrichment_plot(R2_vs_NR2_msvip_ARACNe$mrs_table, R2_vs_NR2_msvip_Regulonlaml$mrs_table)
Make_TF_nes_enrichment_plot(R2_vs_NR2_msvip_Dorothea$mrs_table, R2_vs_NR2_msvip_Regulonlaml$mrs_table)
```

## Analysis R1 vs NR2

```{r}
Make_TF_nes_enrichment_plot(R1_vs_NR2_msvip_ARACNe$mrs_table, R1_vs_NR2_msvip_Dorothea$mrs_table)
Make_TF_nes_enrichment_plot(R1_vs_NR2_msvip_ARACNe$mrs_table, R1_vs_NR2_msvip_Regulonlaml$mrs_table)
Make_TF_nes_enrichment_plot(R1_vs_NR2_msvip_Dorothea$mrs_table, R1_vs_NR2_msvip_Regulonlaml$mrs_table)
```


## Analysis R vs NR

```{r}
Make_TF_nes_enrichment_plot(Clusters_msvip_ARACNe$mrs_table, Clusters_msvip_Dorothea_D$mrs_table)
Make_TF_nes_enrichment_plot(Clusters_msvip_ARACNe$mrs_table, Clusters_msvip_Regulonlaml$mrs_table)
Make_TF_nes_enrichment_plot(Clusters_msvip_Dorothea_D$mrs_table, Clusters_msvip_Regulonlaml$mrs_table)
```

# Mergeing TF activity

## Functions

```{r}
Deal_with_na <- function(df, column, final_value){
  for (i in column){
    data.table::set(df, which(is.na(df[[i]])), i, final_value)
  }
}
```

## Analysis R2 vs NR2

```{r}
TF_1 <- R2_vs_NR2_msvip_ARACNe$mrs_table[,c(1,3:4)] %>%
  dplyr::filter(nes != 0)
colnames(TF_1)[2] <- "nes_1"
colnames(TF_1)[3] <- "pval_1"
TF_2 <- R2_vs_NR2_msvip_Dorothea$mrs_table[,c(1,3:4)] %>%
  dplyr::filter(nes != 0)
colnames(TF_2)[2] <- "nes_2"
colnames(TF_2)[3] <- "pval_2"
TF_3 <- R2_vs_NR2_msvip_Regulonlaml$mrs_table[,c(1,3:4)] %>%
  dplyr::filter(nes != 0)
colnames(TF_3)[2] <- "nes_3"
colnames(TF_3)[3] <- "pval_3"

Mergeing_TF_R2_vs_NR2 <- merge(TF_1, TF_2, by = "TF", all.x = T, all.y = T) %>% 
  merge(TF_3, by = "TF", all.x = T, all.y = T)

Deal_with_na(Mergeing_TF_R2_vs_NR2, c("nes_1", "nes_2", "nes_3"), 0)
Deal_with_na(Mergeing_TF_R2_vs_NR2, c("pval_1", "pval_2", "pval_3"), 1)
```

## Analysis R1 vs NR2

```{r}
TF_1 <- R1_vs_NR2_msvip_ARACNe$mrs_table[,c(1,3:4)] %>%
  dplyr::filter(nes != 0)
colnames(TF_1)[2] <- "nes_1"
colnames(TF_1)[3] <- "pval_1"
# TF_2 <- R1_vs_NR2_msvip_Dorothea$mrs_table[,c(1,3:4)] %>%
#   dplyr::filter(nes != 0)
# colnames(TF_2)[2] <- "nes_2"
# colnames(TF_2)[3] <- "pval_2"
TF_3 <- R1_vs_NR2_msvip_Regulonlaml$mrs_table[,c(1,3:4)] %>%
  dplyr::filter(nes != 0)
colnames(TF_3)[2] <- "nes_3"
colnames(TF_3)[3] <- "pval_3"

Mergeing_TF_R1_vs_NR2 <- merge(TF_1, TF_3, by = "TF", all.x = T, all.y = T) 
# %>% 
#   merge(TF_3, by = "TF", all.x = T, all.y = T)

Deal_with_na(Mergeing_TF_R1_vs_NR2, c("nes_1", "nes_3"), 0)
Deal_with_na(Mergeing_TF_R1_vs_NR2, c("pval_1", "pval_3"), 1)
```

## R vs NR

```{r}
TF_1 <- Clusters_msvip_ARACNe$mrs_table[,c(1,3:4)] %>%
  dplyr::filter(nes != 0)
colnames(TF_1)[2] <- "nes_1"
colnames(TF_1)[3] <- "pval_1"
TF_2 <- Clusters_msvip_ARACNe$mrs_table[,c(1,3:4)] %>%
  dplyr::filter(nes != 0)
colnames(TF_2)[2] <- "nes_2"
colnames(TF_2)[3] <- "pval_2"
TF_3 <- Clusters_msvip_ARACNe$mrs_table[,c(1,3:4)] %>%
  dplyr::filter(nes != 0)
colnames(TF_3)[2] <- "nes_3"
colnames(TF_3)[3] <- "pval_3"

Mergeing_TF_R_vs_NR <- merge(TF_1, TF_2, by = "TF", all.x = T, all.y = T) %>%
  merge(TF_3, by = "TF", all.x = T, all.y = T)

Deal_with_na(Mergeing_TF_R_vs_NR, c("nes_1", "nes_2", "nes_3"), 0)
Deal_with_na(Mergeing_TF_R_vs_NR, c("pval_1", "pval_2", "pval_3"), 1)
```

# Calcul average enrichment and fisher pvalues

## R2 vs NR2

```{r}
Mergeing_TF_R2_vs_NR2$combine_nes <- sapply(1:nrow(Mergeing_TF_R2_vs_NR2), function(TF){
  Sum <- sum(Mergeing_TF_R2_vs_NR2[TF, c(2,4,6)])
  Sum/3
})

Mergeing_TF_R2_vs_NR2$fisher_pval <- sapply(1:nrow(Mergeing_TF_R2_vs_NR2), function(TF){
  pval_1 <- Mergeing_TF_R2_vs_NR2[TF, 3]
  pval_2 <- Mergeing_TF_R2_vs_NR2[TF, 5]
  pval_3 <- Mergeing_TF_R2_vs_NR2[TF, 7]
  poolr::fisher(c(pval_1, pval_2, pval_3), adjust = "none")$p
})

Mergeing_TF_R2_vs_NR2 <- Mergeing_TF_R2_vs_NR2[order(Mergeing_TF_R2_vs_NR2$fisher_pval, decreasing = F),]
Mergeing_TF_R2_vs_NR2 %>% write.table("../Results/Cluster_IDH2/Table/Mergeing_TF_R2_vs_NR2_enrichments.tsv", 
                            sep = "\t", row.names = F, quote = F)
```

## R1 vs NR2

```{r}
Mergeing_TF_R1_vs_NR2$combine_nes <- sapply(1:nrow(Mergeing_TF_R1_vs_NR2), function(TF){
  Sum <- sum(Mergeing_TF_R1_vs_NR2[TF, c(2,4)])
  Sum/2
})

Mergeing_TF_R1_vs_NR2$fisher_pval <- sapply(1:nrow(Mergeing_TF_R1_vs_NR2), function(TF){
  pval_1 <- Mergeing_TF_R1_vs_NR2[TF, 3]
  pval_2 <- Mergeing_TF_R1_vs_NR2[TF, 5]
  poolr::fisher(c(pval_1, pval_2), adjust = "none")$p
})

Mergeing_TF_R1_vs_NR2 <- Mergeing_TF_R1_vs_NR2[order(Mergeing_TF_R1_vs_NR2$fisher_pval, decreasing = F),]
Mergeing_TF_R1_vs_NR2 %>% write.table("../Results/NR_IDH2_R_IDH1/Table/Mergeing_TF_R1_vs_NR2_enrichments.tsv", 
                            sep = "\t", row.names = F, quote = F)
```

## R vs NR

```{r}
Mergeing_TF_R_vs_NR$combine_nes <- sapply(1:nrow(Mergeing_TF_R_vs_NR), function(TF){
  Sum <- sum(Mergeing_TF_R_vs_NR[TF, c(2,4,6)])
  Sum/3
})

Mergeing_TF_R_vs_NR$fisher_pval <- sapply(1:nrow(Mergeing_TF_R_vs_NR), function(TF){
  pval_1 <- Mergeing_TF_R_vs_NR[TF, 3]
  pval_2 <- Mergeing_TF_R_vs_NR[TF, 5]
  pval_3 <- Mergeing_TF_R_vs_NR[TF, 7]
  poolr::fisher(c(pval_1, pval_2), adjust = "none")$p
})

Mergeing_TF_R_vs_NR <- Mergeing_TF_R_vs_NR[order(Mergeing_TF_R_vs_NR$fisher_pval, decreasing = F),]
Mergeing_TF_R_vs_NR %>% write.table("../Results/DGEA_Clusters_Clean/Table/Mergeing_TF_R_vs_NR_enrichments.tsv", 
                            sep = "\t", row.names = F, quote = F)
```

# Volcano plots

## R2 vs NR2

```{r}
p <- EnhancedVolcano(Mergeing_TF_R2_vs_NR2, lab = Mergeing_TF_R2_vs_NR2$TF, x = "combine_nes", y = "fisher_pval", title = "TF_enrichment", pCutoff = 0.1, FCcutoff = 1, labSize = 3.0, ylim = c(0,5), xlim = c(-3,3))

ggsave("../Results/Cluster_IDH2/Volcano_plots/Volcano_Mergeing_TF_R2_vs_NR2_enrichments.png", p)
p
```

## R1 vs NR2

```{r}
p <- EnhancedVolcano(Mergeing_TF_R1_vs_NR2, lab = Mergeing_TF_R1_vs_NR2$TF, x = "combine_nes", y = "fisher_pval", title = "TF_enrichment", pCutoff = 0.1, FCcutoff = 1, labSize = 3.0, ylim = c(0,5), xlim = c(-3,3))

ggsave("../Results/NR_IDH2_R_IDH1/Volcano_plots/Volcano_Mergeing_TF_R1_vs_NR2_enrichments.png", p)
p
```

## R vs NR

```{r}
p <- EnhancedVolcano(Mergeing_TF_R_vs_NR, lab = Mergeing_TF_R_vs_NR$TF, x = "combine_nes", y = "fisher_pval", title = "TF_enrichment", pCutoff = 0.1, FCcutoff = 1, labSize = 3.0, ylim = c(0,5), xlim = c(-3,3))

ggsave("../Results/DGEA_Clusters_Clean/Volcano_plots/Volcano_Mergeing_TF_R_vs_NR_enrichments.png", p)
p
```

# Enrichments

## R2 vs NR2

```{r}
universe <- Mergeing_TF_R2_vs_NR2$TF

TF_up <- Mergeing_TF_R2_vs_NR2 %>%
  dplyr::filter(fisher_pval < 0.1 & combine_nes > 1) %>%
  .$TF

TF_down <- Mergeing_TF_R2_vs_NR2 %>%
  dplyr::filter(fisher_pval < 0.1 & combine_nes < -1) %>%
  .$TF

Enrich_merge_TF_UP <- enrichGO(TF_up, OrgDb = "org.Hs.eg.db", keyType = "SYMBOL", ont = "ALL", pAdjustMethod = "none", universe = universe)

Enrich_merge_TF_DOWN <- enrichGO(TF_down, OrgDb = "org.Hs.eg.db", keyType = "SYMBOL", ont = "ALL", pAdjustMethod = "none", universe = universe)

up <- dotplot(Enrich_merge_TF_UP)  + labs(x = "Enrichment_TF_UP_cluster1_vs_cluster2_mIDH2_merged_network")

ggsave("../Results/Cluster_IDH2/Enrichments/Enrichment_TF_UP_cluster1_vs_cluster2_mIDH2_merged_network.png", up)

down <- dotplot(Enrich_merge_TF_DOWN) + labs(x = "Enrichment_TF_DOWN_cluster1_vs_cluster2_mIDH2_merged_network")

ggsave("../Results/Cluster_IDH2/Enrichments/Enrichment_TF_DOWN_cluster1_vs_cluster2_mIDH2_merged_network.png", down)
```

## R1 vs NR2

```{r}
universe <- Mergeing_TF_R1_vs_NR2$TF

TF_up <- Mergeing_TF_R1_vs_NR2 %>%
  dplyr::filter(fisher_pval < 0.1 & combine_nes > 1) %>%
  .$TF

TF_down <- Mergeing_TF_R1_vs_NR2 %>%
  dplyr::filter(fisher_pval < 0.1 & combine_nes < -1) %>%
  .$TF

Enrich_merge_TF_UP <- enrichGO(TF_up, OrgDb = "org.Hs.eg.db", keyType = "SYMBOL", ont = "ALL", pAdjustMethod = "none", universe = universe)

Enrich_merge_TF_UP@result %>%
  write.table("../Results/NR_IDH2_R_IDH1/Enrichments/Enrichment_TF_UP_cluster1_mIDH1_vs_cluster2_mIDH2_merged_network.tsv", sep = "\t", row.names = F, quote = F)

Enrich_merge_TF_DOWN <- enrichGO(TF_down, OrgDb = "org.Hs.eg.db", keyType = "SYMBOL", ont = "ALL", pAdjustMethod = "none", universe = universe)

Enrich_merge_TF_DOWN@result %>%
  write.table("../Results/NR_IDH2_R_IDH1/Enrichments/Enrichment_TF_DOWN_cluster1_mIDH1_vs_cluster2_mIDH2_merged_network.tsv", sep = "\t", row.names = F, quote = F)

up <- dotplot(Enrich_merge_TF_UP)   + labs(x = "Enrichment_TF_UP_cluster1_mIDH1_vs_cluster2_mIDH2_merged_network")
ggsave("../Results/NR_IDH2_R_IDH1/Enrichments/Enrichment_TF_UP_cluster1_mIDH1_vs_cluster2_mIDH2_merged_network.png", up)

down <- dotplot(Enrich_merge_TF_DOWN)  + labs(x = "Enrichment_TF_DOWN_cluster1_mIDH1_vs_cluster2_mIDH2_merged_network")
ggsave("../Results/NR_IDH2_R_IDH1/Enrichments/Enrichment_TF_DOWN_cluster1_mIDH1_vs_cluster2_mIDH2_merged_network.png", down)
up
down
```

## R vs NR

```{r}
universe <- Mergeing_TF_R_vs_NR$TF

TF_up <- Mergeing_TF_R_vs_NR %>%
  dplyr::filter(fisher_pval < 0.1 & combine_nes > 1) %>%
  .$TF

TF_down <- Mergeing_TF_R_vs_NR %>%
  dplyr::filter(fisher_pval < 0.1 & combine_nes < -1) %>%
  .$TF

Enrich_merge_TF_UP <- enrichGO(TF_up, OrgDb = "org.Hs.eg.db", keyType = "SYMBOL", ont = "ALL", pAdjustMethod = "none", universe = universe)

Enrich_merge_TF_UP@result %>%
  write.table("../Results/DGEA_Clusters_Clean/Enrichments/Enrichment_TF_UP_cluster1_vs_cluster2_merged_network.tsv", sep = "\t", row.names = F, quote = F)

Enrich_merge_TF_DOWN <- enrichGO(TF_down, OrgDb = "org.Hs.eg.db", keyType = "SYMBOL", ont = "ALL", pAdjustMethod = "none", universe = universe)

Enrich_merge_TF_DOWN@result %>%
  write.table("../Results/DGEA_Clusters_Clean/Enrichments/Enrichment_TF_DOWN_cluster1_vs_cluster2_merged_network.tsv", sep = "\t", row.names = F, quote = F)

# up <- dotplot(Enrich_merge_TF_UP) + labs(x = "Enrichment_TF_UP_cluster1_vs_cluster2_merged_network")
# ggsave("../Results/DGEA/Diff_TF_activity/Combined_networks/Enrichments/Enrichment_TF_UP_cluster1_vs_cluster2_merged_network.png", up)

down <- dotplot(Enrich_merge_TF_DOWN)  + labs(x = "Enrichment_TF_DOWN_cluster1_mIDH1_vs_cluster2_mIDH2_merged_network")
ggsave("../Results/DGEA_Clusters_Clean/Enrichments/Enrichment_TF_DOWN_cluster1_vs_cluster2_merged_network.png", down)
up
down
```

```{r}
save.image("/media/alexis/DATA/Session/R_session/Thesis_paper_Rsessions/Gene_exp/Koichi_Different_DGEA_Clusters.RData")
```

