---
title: "Koichi IDHms data analysis"
author: Alexis Hucteau
date: "`r Sys.Date()`"
output:
    html_document:
      toc: yes
      toc_float: yes
      theme: united
editor_options:
  markdown:
    wrap: sentence
---

```{r, error=F}
library(FactoMineR)
library(igraph)
library(limma)
library(viper)
library(stringr)
library(data.table)
library(dplyr)
library(RCy3)
library(factoextra)
library(pheatmap)
library(Hmisc)
library(matrixStats)
library(RColorBrewer)

"%ni%" <- Negate("%in%")
source("~/Core_scripts/msviper_functions.R")
source("~/Core_scripts/core_functions.R")
source("~/GitHub/Multiplex_DNAmet_PPI_Chrom_Coexp/Make_koichi_factor.R")
```

```{r}
RNAseq <- read.csv("~/GitHub/Koichi_gene_expression_analyses_git/Koichi_gene_expression_analyses/DATA/RNAseq_parsed.csv", row.names = 1, header = T, check.names = F)
Clinical_patient_data <- read.table("../Datasets/Clinical_Koichi_data.tsv", sep = "\t", header = T)
Phenotype_IDH <- sapply(colnames(RNAseq), function(sample){
  if(sample %in% Clinical_patient_data$Relapse_RNAseq_data){
    "Relapse"
  }else{
    Clinical_patient_data %>% dplyr::filter(Baseline_RNAseq_data == sample) %>% .$IDH_isoform
  }
})
```

```{r}
RNAseq_Baseline <- RNAseq[Phenotype_IDH != "Relapse"]
Phenotype_IDH <- Phenotype_IDH[Phenotype_IDH != "Relapse"]
```

```{r}
res.pca <- prcomp(t(RNAseq_Baseline))
p <- fviz_pca_ind(res.pca, label="all", habillage=Phenotype_IDH,
                               addEllipses=T, ellipse.level=0.95)
p <- p + ggtitle("PCA gene expression")
p
png("../Results/DGEA/PCA_Koichi_IDH_data.png")
p
dev.off()
```

```{r}
All_sample_ARACNe_network <- read.table("../../Multiplex_DNAmet_PPI_Chrom_Coexp/Output_all_Samples_TPM/network.txt", sep = "\t", header = T)
colnames(All_sample_ARACNe_network)[1:2] <- c("tf", "target")

ref_mIDH1 <- Phenotype_IDH == "IDH1"
ref_mIDH2 <- Phenotype_IDH == "IDH2"

IDHs_Koichi_msvip <- run_msviper(RNAseq_Baseline, All_sample_ARACNe_network, use_aracne = T, ref_mIDH1, ref_mIDH2, "mIDH1", "mIDH2", minsize = 4, ges.filter = T)
Koichi_TF_actitity <- viper(RNAseq_Baseline, dorothea2viper_regulons(IDHs_Koichi_msvip$regulons))
IDHs_Koichi_msvip$mrs_table %>% dplyr::filter(pval < 0.05) %>% write.table("../Results/DGEA/Filtered_TF_Diff_activity_Koichi_IDH.tsv", sep = "\t")
```

```{r}
plot(IDHs_Koichi_msvip$mrs, mrs = 50)
png("../Results/DGEA_Clusters_Clean/Cluster_Koichi_IDH_msviplot.png", width = 1280, height = 720)
plot(IDHs_Koichi_msvip$mrs, mrs = 50)
dev.off()
```

```{r}
DEG_IDH_Koichi <- Differential_analysis(Phenotype_IDH, RNAseq_Baseline)$`IDH1-IDH2`
```

