---
title: "Koichi_IDHm_DNA_methylation_analysis"
author: Alexis Hucteau
date: "`r Sys.Date()`"
output:
    html_document:
      toc: yes
      toc_float: yes
      theme: united
editor_options:
  markdown:
    wrap: sentence
---


```{r}
library(dplyr)
library(stringr)
library(gridExtra)
library(grid)
library(ggplot2)
library(GenomicRanges)
library(RColorBrewer)
library(ChAMP)
library(rpart)
library(rpart.plot)
library(enrichplot)
library(limma)
library(EnhancedVolcano)

source("~/Core_scripts/core_functions.R")
source("~/GitHub/Multiplex_DNAmet_PPI_Chrom_Coexp/Make_koichi_factor.R")
```

```{r}
Clinical_patient_data <- read.table("../Datasets/Clinical_Koichi_data.tsv", sep = "\t", header = T) 
BMIQ <- readRDS("/media/alexis/DATA/Koichi_methylation_dat/BMIQ_norm_Koichi_samples.Rdata")
```

```{r}
B_REL_factor <- colnames(BMIQ) %>% sapply(function(sample){
  sample %in% Clinical_patient_data$Baseline_Sample
})
BMIQ <- BMIQ[,B_REL_factor]
```

```{r}
Pheno_IDHm <- sapply(colnames(BMIQ), function(sample){
  dplyr::filter(Clinical_patient_data, Baseline_Sample == sample) %>% .$IDH_isoform
})
```

```{r}
IDHm_dCpGs <- Differential_analysis(Pheno_IDHm, BMIQ)
IDHm_dCpGs$`IDH1-IDH2`$logFC <- -IDHm_dCpGs$`IDH1-IDH2`$logFC
write.table(IDHm_dCpGs$`IDH1-IDH2`, "../Results/DNAmethylation/dCpGs_KOICHI_IDHm.tsv", sep = "\t", row.names = T, col.names = NA)

BMIQ_no_double <- BMIQ[,Pheno_IDHm != "IDH1_IDH2"]
Pheno_IDHm_no_double <- Pheno_IDHm[Pheno_IDHm != "IDH1_IDH2"]

DMP_IDHm <- champ.DMP(as.matrix(BMIQ_no_double), pheno = Pheno_IDHm_no_double, arraytype = "EPIC", adjPVal = 0.5)
write.table(DMP_IDHm$IDH1_to_IDH2, "../Results/DNAmethylation/DMP_KOICHI_IDHm.tsv", sep = "\t", row.names = T, col.names = NA)

DMR_IDHm <- champ.DMR(as.matrix(BMIQ_no_double), pheno = Pheno_IDHm_no_double, arraytype = "EPIC", cores = 10)
write.table(DMR_IDHm$BumphunterDMR, "../Results/DNAmethylation/DMR_KOICHI_IDHm.tsv", sep = "\t", row.names = T, col.names = NA)

GSEA_KOICHI_IDHm <- ChAMP::champ.GSEA(beta = BMIQ_no_double, DMP = DMP_IDHm$IDH1_to_IDH2, DMR = DMR_IDHm, cores = 10, pheno = Pheno_IDHm, arraytype = "EPIC", adjPval = 1)
GSEA_KOICHI_IDHm$DMP %>% write.table("../Results/DNAmethylation/GSEA_DMP_KOICHI_IDHm.tsv", sep = "\t", row.names = F, quote = F)
GSEA_KOICHI_IDHm$DMR %>% write.table("../Results/DNAmethylation/GSEA_DMR_KOICHI_IDHm.tsv", sep = "\t", row.names = F, quote = F)
```

```{r}
From_GO_to_network <- function(GO_data){
  formatted_genelists <- lapply(1:nrow(GO_data), function(gene_set){
    GO_data[gene_set,]$Genes %>%
      stringr::str_split(pattern = " ") %>%
      unlist %>%
      as.vector %>%
      unique
  })
  lapply(1:(nrow(GO_data)-1), function(gene_set){
    B_start <- gene_set + 1
    lapply(B_start:nrow(GO_data), function(gene_set_B){
      noverlap <- intersect(formatted_genelists[[gene_set]], formatted_genelists[[gene_set_B]]) %>%
        length
      data.frame("A" = GO_data[gene_set, 1], "B" = GO_data[gene_set_B, 1], "nOverlap" = noverlap)
    }) %>%
      data.table::rbindlist()

    }) %>%
    data.table::rbindlist()
}
```

```{r}
Network_GSEA_KOICHI_IDHm <- From_GO_to_network(GSEA_KOICHI_IDHm$DMP)
```

```{r}
Network <- dplyr::filter(Network_GSEA_KOICHI_IDHm, nOverlap > 0)
Network_2 <- dplyr::filter(Network_GSEA_KOICHI_IDHm, nOverlap > 4)

test <- Network[,3] %>% group_by(nOverlap)

test %>% summarize(n = n()) %>%
  ggplot(aes(x = log10(n))) +
  geom_density()
         
Network_2 %>%
  write.table("../Results/DNAmethylation/Enrichments/mIDH_DMP_GSEA_network.tsv", sep = "\t", quote = F, row.names = F)
```


```{r}
p <- EnhancedVolcano(IDHm_dCpGs[["IDH1-IDH2"]], lab = IDHm_dCpGs[["IDH1-IDH2"]]$ID, x = "logFC", y = "P.Value", title = "DMP", pCutoff = 0.05, FCcutoff = 0.1, xlim = c(-0.35,0.35), ylim = c(0,10))

ggsave("../Results/DNAmethylation/Volcano_mIDH1_vs_mIDH2.png", p)
p
```

```{r}
Top_var_CpGs <- rowVars(as.matrix(BMIQ))
names(Top_var_CpGs) <- rownames(BMIQ)

Top_var_CpGs <- Top_var_CpGs[order(Top_var_CpGs, decreasing = T)] %>% .[1:1000000]

Top_Var_CPGs_DMP <- IDHm_dCpGs[["IDH1-IDH2"]] %>% 
  dplyr::filter(ID %in% names(Top_var_CpGs))

p <- EnhancedVolcano(Top_Var_CPGs_DMP, lab = Top_Var_CPGs_DMP$ID, x = "logFC", y = "P.Value", title = "DMP", pCutoff = 0.05, FCcutoff = 0.1, xlim = c(-0.35,0.35), ylim = c(0,10))

ggsave("../Results/DNAmethylation/Volcano_mIDH1_vs_mIDH2_Top_CpGs.png", p)
p
```

