---
title: "Koichi_IDHm_gene_expression_analysis"
author: Alexis Hucteau
date: "`r Sys.Date()`"
output:
    html_document:
      toc: yes
      toc_float: yes
      theme: united
editor_options:
  markdown:
    wrap: sentence
---

```{r}
suppressPackageStartupMessages({
  library(FactoMineR)
  library(igraph)
  library(limma)
  library(viper)
  library(stringr)
  library(data.table)
  library(dplyr)
  library(RCy3)
  library(factoextra)
  library(pheatmap)
  library(Hmisc)
  library(matrixStats)
  library(RColorBrewer)
  library(EnhancedVolcano)
  library(clusterProfiler)
  library(enrichplot)
  library(ggnewscale)
  library(DOSE)
  library(biomaRt)
  library(org.Hs.eg.db)
})

"%ni%" <- Negate("%in%")
source("~/Core_scripts/msviper_functions.R")
source("~/Core_scripts/core_functions.R")
source("~/GitHub/Multiplex_DNAmet_PPI_Chrom_Coexp/Make_koichi_factor.R")
```

```{r}
RNAseq <- read.csv("~/GitHub/Koichi_gene_expression_analyses_git/Koichi_gene_expression_analyses/DATA/RNAseq_parsed.csv", row.names = 1, header = T, check.names = F)

Clinical_patient_data <- read.table("../Datasets/Clinical_Koichi_data.tsv", sep = "\t", header = T)
Phenotype_IDH <- sapply(colnames(RNAseq), function(sample){
  if(sample %in% Clinical_patient_data$Relapse_RNAseq_data){
    "Relapse"
  }else{
    Clinical_patient_data %>% dplyr::filter(Baseline_RNAseq_data == sample) %>% .$IDH_isoform
  }
})
RNAseq <- RNAseq[,Phenotype_IDH %in% c("IDH1", "IDH2")]
Phenotype_IDH <- Phenotype_IDH[Phenotype_IDH %in% c("IDH1", "IDH2")]
write.table(RNAseq, "../Datasets/RNAseq_IDHm_Baseline.tsv", sep = "\t", row.names = F)
```

```{r}
res.pca <- prcomp(t(RNAseq))
p <- fviz_pca_ind(res.pca, label="all", habillage=Phenotype_IDH,
                               addEllipses=T, ellipse.level=0.95)
p <- p + ggtitle("PCA gene expression")
p
png("../Results/DGEA/PCA_Koichi_data_IDHm.png")
p
dev.off()
```

```{r}
Make_heatmap <- function(DATA, Phenotype, method = "pearson", title, annotation_color, kmeans_k = NA, cuttree = NA, corr=T, TF=F) {
  annotation_for_heatmap <- data.frame(Phenotype = Phenotype$Phenotype)
  rownames(annotation_for_heatmap) <- colnames(DATA)
  if(corr){
    corr <- rcorr(as.matrix(DATA), type = method)$r
    colnames(corr) <- colnames(DATA)
    if(TF){
      color= rev(colorRampPalette(brewer.pal(n = 9, name = "RdBu"))(100))
    }else{
      color = colorRampPalette(brewer.pal(n = 9, name = "YlOrRd"))(100)
    }
  }else{
    corr=DATA
    color= rev(colorRampPalette(brewer.pal(n = 9, name = "RdBu"))(100))
  }
  
  title <- paste0(title, " ", method)
  heatmap <- pheatmap(corr, 
                      color = color,
                      annotation_col = annotation_for_heatmap,
                      annotation_colors = annotation_color,
                      legend = TRUE,
                      treeheight_row = 20,
                      main = title, 
                      fontsize = 10,
                      cutree_cols = cuttree
  )
  return(heatmap)
}
```

```{r}
Var_gene <- rowVars(as.matrix(RNAseq))
names(Var_gene) <- rownames(RNAseq)
```

```{r}
nb_genes <- 100
method <- "pearson"
Top_var_genes <- Var_gene[order(Var_gene, decreasing = T)] %>% .[1:nb_genes]
Top_Var_RNAseq <- RNAseq %>% .[rownames(.) %in% names(Top_var_genes),] 
ann_color_RNA <- list(
    Phenotype = c(IDH1 = "blue", IDH2 = "red"))

pheno_heatmap <- data.frame("IDH1" = Phenotype_IDH == "IDH1", "IDH2" = Phenotype_IDH == "IDH2")
pheno_heatmap$Phenotype <- Phenotype_IDH
heat <- Make_heatmap(Top_Var_RNAseq, Phenotype = pheno_heatmap, title = paste0("Patient correlation of Top ", nb_genes, " variable Gene expression"), annotation_color = ann_color_RNA, method = method)

png(paste0("../Results/DGEA/Heatmap_IDHm_GE_top_", nb_genes,"_corr_", method, "_Koichi_data.png"))
heat
dev.off()

res.pca <- prcomp(t(Top_Var_RNAseq))
p <- fviz_pca_ind(res.pca, label="all", habillage=Phenotype_IDH,
                               addEllipses=T, ellipse.level=0.95)
p <- p + ggtitle(paste0("PCA top ", nb_genes, " genes"))
p
png(paste0("../Results/DGEA/PCA_IDHm_GE_top_", nb_genes,"_Koichi_data.png"))
p
dev.off()
if(nb_genes < 500){
  heat2 <- Make_heatmap(Top_Var_RNAseq, Phenotype = pheno_heatmap, title = paste0("Patient Clusters correlation of Top ", nb_genes, " variable Gene expression"), annotation_color = ann_color_RNA, method = "", corr = F)
  png(paste0("../Results/DGEA/Heatmap_IDHm_GE_top_", nb_genes,"_Koichi_data.png"))
  heat2
  dev.off()
}
```

```{r}
All_sample_ARACNe_network <- read.table("../../Multiplex_DNAmet_PPI_Chrom_Coexp/Output_all_Samples_TPM/network.txt", sep = "\t", header = T)
colnames(All_sample_ARACNe_network)[1:2] <- c("tf", "target")

ref_IDH1 <- Phenotype_IDH == "IDH1"
ref_IDH2 <- Phenotype_IDH == "IDH2"

IDHm_msvip <- run_msviper(RNAseq, All_sample_ARACNe_network, use_aracne = T, ref_IDH2, ref_IDH1, "IDH2", "IDH1", minsize = 4, ges.filter = T)
Koichi_TF_actitity <- viper(RNAseq, dorothea2viper_regulons(IDHm_msvip$regulons))
```
```{r}
regulonlaml <- read.table("../Datasets/regulonaml_SYMBOL.tsv", sep = "\t", header = T)
colnames(regulonlaml)[1:2] <- c("tf", "target")

ref_IDH1 <- Phenotype_IDH == "IDH1"
ref_IDH2 <- Phenotype_IDH == "IDH2"

IDHm_regulonlaml_msvip <- run_msviper(RNAseq, All_sample_ARACNe_network, use_aracne = T, ref_IDH2, ref_IDH1, "IDH2", "IDH1", minsize = 4, ges.filter = T)
Koichi_regulonlaml_TF_actitity <- viper(RNAseq, dorothea2viper_regulons(IDHm_regulonlaml_msvip$regulons))
```



```{r}
png("../Results/DGEA_IDHm_msVIPER/Koichi_IDHm_msviplot.png", width = 1280, height = 720)
plot(IDHm_msvip$mrs, mrs = 50)
dev.off()
plot(IDHm_msvip$mrs, mrs = 50)
IDHm_msvip$mrs_table %>% 
  write.table("../Results/DGEA/TF_Diff_activity_IDHm_KOICHI.tsv", sep="\t", row.names = F)
IDHm_msvip$mrs_table %>% dplyr::filter(pval < 0.1) %>% 
  write.table("../Results/DGEA/Filtered_TF_Diff_activity_IDHm_KOICHI.tsv", sep="\t", row.names = F)
IDHm_msvip$regulons %>% 
  write.table("../Results/DGEA/IDHm_KOICHI_GRN.tsv", sep = "\t", row.names = F)
```

```{r}
png("../Results/DGEA_IDHm_msVIPER/Koichi_IDHm_regulonlaml_msviplot.png", width = 1280, height = 720)
plot(IDHm_regulonlaml_msvip$mrs, mrs = 50)
dev.off()
plot(IDHm_regulonlaml_msvip$mrs, mrs = 50)
IDHm_regulonlaml_msvip$mrs_table %>% 
  write.table("../Results/DGEA/TF_Diff_activity_IDHm_regulonlaml_KOICHI.tsv", sep="\t", row.names = F)
IDHm_regulonlaml_msvip$mrs_table %>% dplyr::filter(pval < 0.1) %>% 
  write.table("../Results/DGEA/Filtered_TF_Diff_activity_IDHm_regulonlaml_KOICHI.tsv", sep="\t", row.names = F)
IDHm_regulonlaml_msvip$regulons %>% 
  write.table("../Results/DGEA/IDHm_KOICHI_GRN_regulonlaml.tsv", sep = "\t", row.names = F)
```



```{r}
Var_TF <- rowVars(as.matrix(Koichi_TF_actitity))
names(Var_TF) <- rownames(Koichi_TF_actitity)
```

```{r}
nb_TF <- 5
method <- "pearson"
Top_var_TF <- Var_TF[order(Var_TF, decreasing = T)] %>% .[1:nb_TF]
Top_Var_TF <- Koichi_TF_actitity %>% .[rownames(.) %in% names(Top_var_TF),] 
heat <- Make_heatmap(Top_Var_TF, Phenotype = pheno_heatmap, title = paste0("Patient Clusters correlation of Top ", nb_TF, " variable TF"), annotation_color = ann_color_RNA, method = method,TF=T)

png(paste0("../Results/DGEA/Heatmap_Clusters_TF_top_", nb_TF,"_corr_", method, "_Koichi_data.png"))
heat
dev.off()

res.pca <- Top_Var_TF %>% as.data.frame %>% t() %>% prcomp

p <- fviz_pca_ind(res.pca, label="all", habillage=Phenotype_Cluster,
                               addEllipses=T, ellipse.level=0.95)
p <- p + ggtitle(paste0("PCA Clusters top ", nb_TF, " TF"))
p
png(paste0("../Results/DGEA/PCA_Clusters_GE_top_", nb_TF,"_TF_Koichi_data.png"))
p
dev.off()
heat2 <- Make_heatmap(Top_Var_TF, Phenotype = pheno_heatmap, title = paste0("Patient Clusters correlation of Top ", nb_TF, " variable TF"), annotation_color = ann_color_RNA, method = "", corr=F)

png(paste0("../Results/DGEA/Heatmap_Clusters_TF_top_", nb_TF,"_Koichi_data.png"))
heat2
dev.off()
```

```{r}
res.pca <- Koichi_TF_actitity %>% as.data.frame %>% prcomp

p <- fviz_pca_ind(res.pca, label="all",
                               addEllipses=F, ellipse.level=0.95)
p <- p + ggtitle("PCA TF activity")
p
png("../Results/DGEA/Cluster_PCA_TF_Koichi_data.png")
p
dev.off()
```

```{r}
Cluster_DEG_analysis <- Differential_analysis(pheno_heatmap$Phenotype, RNAseq)$`Cluster_NR-Cluster_R`
Cluster_DEG_analysis %>% dplyr::filter(abs(logFC) > 1.5 & P.Value < 0.1) %>%
  write.table("../Results/DGEA/Cluster_filtered_DEG_analysis.tsv", sep="\t", col.names = NA)
```

```{r}
Volcano <- EnhancedVolcano(Cluster_DEG_analysis, 
                           lab = rownames(Cluster_DEG_analysis), 
                           x = 'logFC', y = 'P.Value', 
                           FCcutoff = 0.75, pCutoff = 0.05,
                           title="Cluster DGE", 
                           shape = 1)
Volcano
png("../Results/DGEA/Clusters_DEG_volcanoplot.png")
Volcano
dev.off()
```

```{r}
Gene_Up <- Cluster_DEG_analysis %>% dplyr::filter(P.Value < 0.05 & logFC > 1.5) %>% .$ID
Gene_Down <- Cluster_DEG_analysis %>% dplyr::filter(P.Value < 0.05 & logFC < - 1.5) %>% .$ID
GO_UP_enrich_cluster <- enrichGO(gene = Gene_Up, OrgDb = "org.Hs.eg.db", keyType = "SYMBOL")
Go_Down_enrich_cluster <- enrichGO(gene = Gene_Down, OrgDb = "org.Hs.eg.db", keyType = "SYMBOL")

DEGenes <- Cluster_DEG_analysis %>% dplyr::filter(P.Value < 0.05 & abs(logFC) > 1.5) %>% .$logFC
names(DEGenes) <- Cluster_DEG_analysis %>% dplyr::filter(P.Value < 0.05 & abs(logFC) > 1.5) %>% .$ID

DEGenes = sort(DEGenes, decreasing = TRUE)
gse <- gseGO(geneList=DEGenes, 
             ont ="ALL", 
             keyType = "SYMBOL", 
             nPerm = 10000, 
             minGSSize = 3, 
             maxGSSize = 800, 
             pvalueCutoff = 0.05, 
             verbose = TRUE, 
             OrgDb = "org.Hs.eg.db", 
             pAdjustMethod = "none")


png("../Results/DGEA/Clusters_GO.png")
dotplot(gse, title = "GO enrichment",showCategory=20, split=".sign") + facet_grid(.~.sign)
dev.off()
dotplot(gse, title = "GO enrichment",showCategory=20, split=".sign") + facet_grid(.~.sign)

hs <- org.Hs.eg.db
DEGenes_ID <- select(hs, names(DEGenes), columns = c("ENTREZID", "SYMBOL"), keytype = "SYMBOL") %>% na.omit() %>% .[!duplicated(.$SYMBOL),]
DEGenes_ID_list <- DEGenes[which(names(DEGenes) %in% DEGenes_ID$SYMBOL)]
names(DEGenes_ID_list) <- DEGenes_ID$ENTREZID

dse <- gseDO(DEGenes_ID_list, pAdjustMethod = "none", pvalueCutoff = 0.1, minGSSize = 5)
ridgeplot(dse)
dotplot(dse, title = "DO enrichment", showCategory=20, split=".sign") + facet_grid(.~.sign)
png("../Results/DGEA/Ridgeplot_DO.png")
ridgeplot(dse)
dev.off()
png("../Results/DGEA/Dot_DO.png")
dotplot(dse, title = "DO enrichment", showCategory=20, split=".sign") + facet_grid(.~.sign)
dev.off()
kse <- gseKEGG(DEGenes_ID_list, minGSSize = 5, pvalueCutoff = 0.1, pAdjustMethod = "none")
dotplot(kse,showCategory=20, title = "KEGG enrichment",  split=".sign") + facet_grid(.~.sign)
png("../Results/DGEA/Dot_KEGG.png")
dotplot(kse, showCategory=20, title = "KEGG enrichment", split=".sign") + facet_grid(.~.sign)
dev.off()
```

```{r}
gse@result %>% write.table("../Results/DGEA/gse_Cluster.tsv", sep = "\t", col.names = NA)
kse@result %>% write.table("../Results/DGEA/kse_Cluster.tsv", sep = "\t", col.names = NA)
dse@result %>% write.table("../Results/DGEA/dse_Cluster.tsv", sep = "\t", col.names = NA)
```

```{r}
enrichGO_UP <- enrichGO(Gene_Up, OrgDb = "org.Hs.eg.db", keyType = "SYMBOL", pvalueCutoff = 0.05, pAdjustMethod = "none", minGSSize = 5)
dotplot(enrichGO_UP,showCategory=15, title = "GO genes up enrichment")
enrichGO_down <- enrichGO(Gene_Down, OrgDb = "org.Hs.eg.db", keyType = "SYMBOL", pvalueCutoff = 0.05, pAdjustMethod = "none", minGSSize = 5)
dotplot(enrichGO_down,showCategory=15, title = "GO genes down enrichment")
goplot(enrichGO_UP)
png("../Results/DGEA/Goplot_Up.png")
goplot(enrichGO_UP)
dev.off()
goplot(enrichGO_down)
png("../Results/DGEA/Goplot_Down.png")
goplot(enrichGO_down)
dev.off()
kse_readable <- setReadable(kse, 'org.Hs.eg.db', "ENTREZID")
p1 <- cnetplot(kse_readable, showCategory = 20, foldChange = DEGenes_ID_list, colorEdge=T)

p1 <- cnetplot(kse_readable, node_label="category", showCategory = 20, 
        cex_label_category = 1.2, foldChange = DEGenes_ID_list, colorEdge=T) 
p2 <- cnetplot(kse_readable, node_label="gene", showCategory = 20, 
        cex_label_gene = 0.8, colorEdge=T) 
p3 <- cnetplot(kse_readable, node_label="none", showCategory = 20, 
        color_category='firebrick', 
        color_gene='steelblue', colorEdge=T) 
png("../Results/DGEA/KSE_cowplot.png", height = 1080, width = 1920)
cowplot::plot_grid(p1, p2, p3, ncol=2, labels=LETTERS[1:3])
dev.off()
p1 <- cnetplot(kse_readable, node_label="all", showCategory = 20, foldChange = DEGenes_ID_list, color_category='firebrick', colorEdge=T) 
p2 <- cnetplot(dse_readable, node_label="all", showCategory = 20, foldChange = DEGenes_ID_list, color_category='firebrick', colorEdge=T) 
p3 <- cnetplot(gse, node_label="all", showCategory = 20, foldChange = DEGenes_ID_list, color_category='firebrick', colorEdge=T) 

png("../Results/DGEA/Cluster_cowplot.png", height = 1080, width = 1920)
cowplot::plot_grid(p1, p2, p3, ncol=2, labels=c("KEGG", "Disease Ontology", "Gene Ontology"))
dev.off()
p2 <- heatplot(dse_readable, foldChange=DEGenes_ID_list, showCategory=10)
p2
edox2 <- pairwise_termsim(dse_readable, showCategory = 200)
p1 <- treeplot(edox2)
p2 <- treeplot(edox2, hclust_method = "average")
aplot::plot_list(p1, p2, tag_levels='A')
png("../Results/DGEA/DSE_treeplot.png", height = 1080, width = 1920)
aplot::plot_list(p1, p2, tag_levels='A')
dev.off()
```
