---
title: "TUH Patients IDHms data analysis"
author: Alexis Hucteau
date: "`r Sys.Date()`"
output:
    html_document:
      toc: yes
      toc_float: yes
      theme: united
editor_options:
  markdown:
    wrap: sentence
---

```{r, error=F}
library(FactoMineR)
library(igraph)
library(limma)
library(viper)
library(stringr)
library(data.table)
library(dplyr)
library(RCy3)
library(factoextra)
library(pheatmap)
library(Hmisc)
library(matrixStats)
library(RColorBrewer)
library(sva)
library(aracne.networks)
library(GeneAnswers)
library(org.Hs.eg.db)


"%ni%" <- Negate("%in%")
source("~/Core_scripts/msviper_functions.R")
source("~/GitHub/Multiplex_DNAmet_PPI_Chrom_Coexp/Make_koichi_factor.R")
source("~/Core_scripts/core_functions.R")
```

```{r}
RNAseq_TUH <- read.table("../Datasets/RNASEQ_FULL_TABLE_SARRY.tsv", sep = "\t", header = T, row.names = 1) 
RNAseq_TUH_all_patients <- read.table("../Datasets/FEATURECOUNT_RAW_RNASEQ_FULL_TABLE_SARRY.gct", sep = "\t", header = T, row.names = 2, skip = 2, check.names = F)
RNAseq_TUH_all_patients <- RNAseq_TUH_all_patients[-1]

TUH_Clinicals_2 <- read.delim("~/GitHub/Thesis_paper/Datasets/TUH_Clinicals_2.tsv")

Clinical_data_2_Keep <- TUH_Clinicals_2[c(1:5, 81:82, 10, 17:19, 23, 27)]

TUH_PATIENTS_CORRESPONDANCES <- read.table("../Datasets/CORRESPONDANCE_PATIENTS_TOULOUSE.tsv", sep = "\t", header = T)
TUH_PATIENTS_CORRESPONDANCES$RNAseq_ID <- paste(TUH_PATIENTS_CORRESPONDANCES$PATIENT.ACRONYME, TUH_PATIENTS_CORRESPONDANCES$BATCH, sep = "_")

RNAseq_TUH_all_patients <- RNAseq_TUH_all_patients %>% .[colnames(.) %in% TUH_PATIENTS_CORRESPONDANCES$NAME.RNAseq]

Clinical_data_2_Keep$Final_data_ID <- sapply(1:nrow(Clinical_data_2_Keep), function(id){
  Site <- Clinical_data_2_Keep[id, "Clinical.site"]
  ID <- Clinical_data_2_Keep[id, "Patient.ID"]
  if(Site == "TUH"){
    ID <- as.numeric(ID)
    TUH_ID <- ifelse(ID < 10, paste0("0", ID), ID)
    final.id <- paste0(Site, TUH_ID)
  }else{
    final.id <- ID
  }
  final.id
})
Clinical_data_2_Keep <- dplyr::filter(Clinical_data_2_Keep, "Final_data_ID" != "TUHNA")

Clinical_data_2_Keep_RNAseq <- merge(Clinical_data_2_Keep, TUH_PATIENTS_CORRESPONDANCES, by.x = "Final_data_ID", by.y = "PATIENT.ACRONYME")
```

```{r}
RNAseq_TUH_all_patients <- RNAseq_TUH_all_patients[colnames(RNAseq_TUH_all_patients) %in% Clinical_data_2_Keep_RNAseq$NAME.RNAseq]

Batches_Transcripto <- sapply(colnames(RNAseq_TUH_all_patients), function(sample){
  Clinical_data_2_Keep_RNAseq %>% dplyr::filter(NAME.RNAseq == sample) %>% .$BATCH
}) %>% unlist
RNAseq_TUH_all_patients <- na.omit(RNAseq_TUH_all_patients)
```

```{r}
res.pca <- prcomp(t(RNAseq_TUH_all_patients))
p <- fviz_pca_ind(res.pca, label="all", habillage=Batches_Transcripto,
                               addEllipses=T, ellipse.level=0.95)
p <- p + ggtitle("PCA gene expression")
p
png("../Results/DGEA/PCA_TUH_data.png")
p
dev.off()

TUH_batch_effect_removed <- ComBat(RNAseq_TUH_all_patients, Batches_Transcripto)

res.pca <- prcomp(t(TUH_batch_effect_removed))
p <- fviz_pca_ind(res.pca, label="all", habillage=Batches_Transcripto,
                               addEllipses=T, ellipse.level=0.95)
p <- p + ggtitle("PCA gene expression")
p
png("../Results/DGEA/PCA_TUH_Batch_removed_data.png")
p
dev.off()

TUH_batch_effect_removed %>% 
  write.table("../Datasets/TUH_batch_effect_removed.tsv", sep = "\t")

```
```{r}
Pheno_TUH_mutation <- function(mutation){
  sapply(colnames(TUH_batch_effect_removed), function(sample){
  patient_data <- dplyr::filter(Clinical_data_2_Keep_RNAseq, NAME.RNAseq == sample)
  Test_mutation <- stringr::str_detect(patient_data$Genetic.abnormality.at.initial.diagnostic..clinically.validated., pattern = mutation)
  ifelse(Test_mutation, mutation, "wt")
  })
}

Pheno_IDH_mutation <- sapply(colnames(TUH_batch_effect_removed), function(sample){
  patient_data <- dplyr::filter(Clinical_data_2_Keep_RNAseq, NAME.RNAseq == sample)
  if(patient_data$Clinical.site !="UPenn"){
    Test_IDH1 <- stringr::str_detect(patient_data$Genetic.abnormality.at.initial.diagnostic..clinically.validated., pattern = "IDH1")
    Test_IDH2 <- stringr::str_detect(patient_data$Genetic.abnormality.at.initial.diagnostic..clinically.validated., pattern = "IDH2")
    Test_IDH1 <- ifelse(Test_IDH1, "mIDH1", "")
    Test_IDH2 <- ifelse(Test_IDH2, "mIDH2", "")
    res <- paste(Test_IDH1, Test_IDH2, sep = "_")
    res <- ifelse(res != "mIDH1_mIDH2", stringr::str_remove(res, "_"), res)
  }else{
    Test_IDH <- stringr::str_detect(patient_data$Genetic.abnormality.at.initial.diagnostic..clinically.validated., pattern = "IDH")
    res <- ifelse(Test_IDH, "NA", "")
  }
    ifelse(res == "", "wtIDH", res)
  })
Pheno_IDH_mutation %>% table

data.frame(Sample = colnames(TUH_batch_effect_removed), Pheno_IDH = Pheno_IDH_mutation) %>% write.table("../Datasets/Pheno_IDH_mutation_TUH_allsamples.tsv", sep = "\t", col.names = NA)
```

```{r}
p <- fviz_pca_ind(res.pca, label="all", habillage=Pheno_IDH_mutation,
                               addEllipses=T, ellipse.level=0.95)
p <- p + ggtitle("PCA gene expression")
p
png("../Results/DGEA/PCA_TUH_IDH_data.png")
p
dev.off()
```
```{r}
data(regulonlaml)
viper_regulons2dorothea <- function(r) {
  res <- r %>%
    purrr::map_df(
      .f = function(i) {
        tf_target <- i$tfmode %>%
          tibble::enframe(name = "target", value = "mor") %>%
          mutate(likelihood = i$likelihood)
      },
      .id = "tf"
    )
  return(res)
}

regulonaml <- viper_regulons2dorothea(regulonlaml)
colnames(regulonaml)[1:2] <- c("tf", "target")
regulonaml_SYMBOL <- data.frame("tf" = GeneAnswers::getSymbols(regulonaml$tf, data = "org.Hs.eg"),
                                "target" = GeneAnswers::getSymbols(regulonaml$target, data = "org.Hs.eg"),
                                "mor" = regulonaml$mor,
                                "likelihood" = regulonaml$likelihood)
```


```{r}
ref_mIDH1 <- Pheno_IDH_mutation == "mIDH1"
ref_mIDH2 <- Pheno_IDH_mutation == "mIDH2"

IDHs_TUH_msvip <- run_msviper(TUH_batch_effect_removed, regulonaml_SYMBOL, use_aracne = T, ref_mIDH1, ref_mIDH2, "mIDH1", "mIDH2", minsize = 4, ges.filter = T)
TUH_TF_actitity <- viper(TUH_batch_effect_removed, dorothea2viper_regulons(IDHs_TUH_msvip$regulons))
IDHs_TUH_msvip$mrs_table %>% dplyr::filter(pval < 0.05) %>% write.table("../Results/DGEA/Filtered_TF_Diff_activity_TUH_IDH_regulonlaml.tsv", sep = "\t")
```

```{r}
png("../Results/DGEA_Clusters_Clean/IDHs_TUH_msvip.png", width = 1280, height = 720)
plot(IDHs_TUH_msvip$mrs, mrs = 50)
dev.off()
```



```{r}
DEG_IDH_TUH <- Differential_analysis(Pheno_IDH_mutation, TUH_batch_effect_removed)
DEG_IDH_TUH$`mIDH1-mIDH2` %>% write.table("../Results/DGEA/DEGs_tables/DEG_IDH_TUH.tsv", sep = "\t", col.names = NA)
DEG_IDH_TUH$`mIDH1-wtIDH` %>% write.table("../Results/DGEA/DEGs_tables/DEG_IDH1_wt_TUH.tsv", sep = "\t", col.names = NA)
DEG_IDH_TUH$`mIDH2-wtIDH` %>% write.table("../Results/DGEA/DEGs_tables/DEG_IDH2_wt_TUH.tsv", sep = "\t", col.names = NA)
```









